diff --git a/apollos-church-api/.gitignore b/apollos-church-api/.gitignore
index 30117522..93c0f8f5 100644
--- a/apollos-church-api/.gitignore
+++ b/apollos-church-api/.gitignore
@@ -97,4 +97,5 @@ packages/**/docs/generated/*
 # shared secrets
 .env.shared
 
-development.db
\ No newline at end of file
+# development database
+development.db
diff --git a/apollos-church-api/apollos.json b/apollos-church-api/apollos.json
index 635d0fe0..e82a4e48 100644
--- a/apollos-church-api/apollos.json
+++ b/apollos-church-api/apollos.json
@@ -1,4 +1,4 @@
 {
-  "version": "2.2.0",
+  "version": "2.3.0",
   "environment": "api"
 }
diff --git a/apollos-church-api/config.yml b/apollos-church-api/config.yml
index 33dec71c..84013a9e 100644
--- a/apollos-church-api/config.yml
+++ b/apollos-church-api/config.yml
@@ -26,7 +26,7 @@ BUGSNAG:
   ## Optional, but you'll need to remove references to @apollosproject/bugsnag if you don't wish to use.
   API_KEY: ${BUGSNAG_KEY}
 ROCK:
-  API_URL: https://rock.apollos.app/api
+  API_URL: ${ROCK_API}
   API_TOKEN: ${ROCK_TOKEN}
   IMAGE_URL: https://rock.apollos.app/GetImage.ashx
   SHARE_URL: https://rock.apollos.app
@@ -34,6 +34,7 @@ ROCK:
   TIMEZONE: 'America/New_York'
   USE_PLUGIN: true
   SHOW_INACTIVE_CONTENT: ${ROCK_SHOW_INACTIVE_CONTENT}
+  VERSION: 10
 UNIVERSAL_LINKS:
   # This should be your app's bundle ID, as found in your info.plist.
   APPLE_APP_ID: ${APPLE_APP_ID}
@@ -47,12 +48,17 @@ UNIVERSAL_LINKS:
   # https://docs.getsocial.im/knowledge-base/android-signing-key-sha256/
   GOOGLE_KEYSTORE_SHA256: ${GOOGLE_KEYSTORE_SHA256}
 ONE_SIGNAL:
-  APP_ID: b6e75a77-003f-4466-95ca-82cc5cdc407b
+  APP_ID: ${ONE_SIGNAL_APP_ID}
   REST_KEY: ${ONE_SIGNAL_REST_KEY}
 CHURCH_ONLINE:
-  URL: https://apollos.online.church/
+  URL: ${CHURCH_ONLINE_URL}
   MEDIA_URLS: []
   WEB_VIEW_URL: https://apollos.online.church/
+SUGGESTED_FOLLOWS:
+  - michael.neeley@differential.com
+  - 
+    email: vince@classyh.at
+    campusId: 45893d11-cd72-4a42-b3ee-08814a937e1a
 ALGOLIA:
   APPLICATION_ID: ${ALGOLIASEARCH_APPLICATION_ID}
   API_KEY: ${ALGOLIASEARCH_API_KEY}
@@ -86,7 +92,7 @@ ANALYTICS:
   # on the accompanying service
   SEGMENT_KEY:
   GA_ID:
-  USE_ROCK: true
+  USE_ROCK: false
 BIBLE_API:
   KEY: ${BIBLE_API_KEY}
   BIBLE_ID:
@@ -134,12 +140,14 @@ ROCK_MAPPINGS:
     # Id of the category used to store all persona dataviews
     PersonaId: 210
 
-  # Apollos will pull these content channels and expose them in the app on the discover screen.
+  # TODO deprecated
+  # used for the contentChannels root level query
   DISCOVER_CONTENT_CHANNEL_IDS:
     - 3
     - 5
     - 6
     - 8
+
   # Apollos will pull content items from these channels and expose them on the user feed
   # However, it will not expose the channel itself.
   # These channels are also pulled when indexing searchable content.
diff --git a/apollos-church-api/package.json b/apollos-church-api/package.json
index 3e0a4dd8..b7ad37c2 100644
--- a/apollos-church-api/package.json
+++ b/apollos-church-api/package.json
@@ -20,22 +20,22 @@
     "node": ">=12.x.x"
   },
   "dependencies": {
-    "@apollosproject/bugsnag": "^2.2.0",
-    "@apollosproject/config": "^2.2.0",
-    "@apollosproject/data-connector-algolia-search": "^2.2.0",
-    "@apollosproject/data-connector-analytics": "^2.2.0",
-    "@apollosproject/data-connector-bible": "^2.2.0",
-    "@apollosproject/data-connector-church-online": "^2.2.0",
-    "@apollosproject/data-connector-cloudinary": "^2.2.0",
-    "@apollosproject/data-connector-onesignal": "^2.2.0",
-    "@apollosproject/data-connector-passes": "^2.2.0",
-    "@apollosproject/data-connector-postgres": "^2.2.0",
-    "@apollosproject/data-connector-redis-cache": "^2.2.0",
-    "@apollosproject/data-connector-rock": "^2.2.0",
-    "@apollosproject/data-connector-twilio": "^2.2.0",
-    "@apollosproject/data-schema": "^2.2.0",
-    "@apollosproject/rock-apollo-data-source": "^2.2.0",
-    "@apollosproject/server-core": "^2.2.0",
+    "@apollosproject/bugsnag": "^2.3.1",
+    "@apollosproject/config": "^2.3.1",
+    "@apollosproject/data-connector-algolia-search": "^2.3.1",
+    "@apollosproject/data-connector-analytics": "^2.3.1",
+    "@apollosproject/data-connector-bible": "^2.3.1",
+    "@apollosproject/data-connector-church-online": "^2.3.1",
+    "@apollosproject/data-connector-cloudinary": "^2.3.1",
+    "@apollosproject/data-connector-onesignal": "^2.3.1",
+    "@apollosproject/data-connector-passes": "^2.3.1",
+    "@apollosproject/data-connector-postgres": "^2.3.1",
+    "@apollosproject/data-connector-redis-cache": "^2.3.1",
+    "@apollosproject/data-connector-rock": "^2.3.1",
+    "@apollosproject/data-connector-twilio": "^2.3.1",
+    "@apollosproject/data-schema": "^2.3.1",
+    "@apollosproject/rock-apollo-data-source": "^2.3.1",
+    "@apollosproject/server-core": "^2.3.1",
     "apollo-server": "^2.14.2",
     "apollo-server-express": "2.14.2",
     "color": "3.0.0",
@@ -48,9 +48,9 @@
     "sqlite3": "^5.0.1"
   },
   "devDependencies": {
-    "@apollosproject/apollo-server-env-mock": "^2.2.0",
-    "@apollosproject/babel-preset-apollos": "^2.2.0",
-    "@apollosproject/eslint-config": "^2.2.0",
+    "@apollosproject/apollo-server-env-mock": "^2.3.1",
+    "@apollosproject/babel-preset-apollos": "^2.3.1",
+    "@apollosproject/eslint-config": "^2.3.1",
     "@babel/cli": "^7.4.5",
     "@babel/core": "^7.4.5",
     "@babel/node": "^7.7.4",
diff --git a/apollos-church-api/src/data/index.js b/apollos-church-api/src/data/index.js
index 4eb7c722..7f73dd94 100644
--- a/apollos-church-api/src/data/index.js
+++ b/apollos-church-api/src/data/index.js
@@ -40,6 +40,7 @@ import {
 import {
   Comment,
   UserFlag,
+  Follow,
   Campus as PostgresCampus,
   Person as PostgresPerson,
 } from '@apollosproject/data-connector-postgres';
@@ -91,6 +92,7 @@ const data = {
   PrayerRequest,
   Comment,
   UserFlag,
+  Follow,
   PostgresCampus,
   Persona,
   Person, // An extension of Postgres person. Will be eliminated in the near future so you can use just postgres/Person.
