diff --git a/apolloschurchapp/apollos.json b/apolloschurchapp/apollos.json
index 38517b2..c73c48a 100644
--- a/apolloschurchapp/apollos.json
+++ b/apolloschurchapp/apollos.json
@@ -1,4 +1,4 @@
 {
-    "version": "1.2.0-beta.7",
+    "version": "1.2.0-beta.6",
     "environment": "client"
 }
\ No newline at end of file
diff --git a/apolloschurchapp/package.json b/apolloschurchapp/package.json
index 20d4fe7..308aa43 100644
--- a/apolloschurchapp/package.json
+++ b/apolloschurchapp/package.json
@@ -1,6 +1,6 @@
 {
   "name": "apolloschurchapp",
-  "version": "1.2.0-beta.7",
+  "version": "1.2.0-beta.6",
   "private": true,
   "devDependencies": {
     "codecov": "3.0.2",
@@ -63,16 +63,16 @@
   },
   "dependencies": {
     "@apollosproject/react-native-airplay-btn": "0.2.0",
-    "@apollosproject/ui-analytics": "^1.2.0-beta.7",
-    "@apollosproject/ui-auth": "^1.2.0-beta.7",
-    "@apollosproject/ui-htmlview": "^1.2.0-beta.7",
-    "@apollosproject/ui-kit": "^1.2.0-beta.7",
-    "@apollosproject/ui-media-player": "^1.2.0-beta.7",
-    "@apollosproject/ui-notifications": "^1.2.0-beta.7",
-    "@apollosproject/ui-onboarding": "^1.2.0-beta.7",
-    "@apollosproject/ui-passes": "^1.2.0-beta.7",
-    "@apollosproject/ui-scripture": "^1.2.0-beta.7",
-    "@apollosproject/ui-storybook": "^1.2.0-beta.7",
+    "@apollosproject/ui-analytics": "^1.2.0-beta.6",
+    "@apollosproject/ui-auth": "^1.2.0-beta.6",
+    "@apollosproject/ui-htmlview": "^1.2.0-beta.6",
+    "@apollosproject/ui-kit": "^1.2.0-beta.6",
+    "@apollosproject/ui-media-player": "^1.2.0-beta.6",
+    "@apollosproject/ui-notifications": "^1.2.0-beta.6",
+    "@apollosproject/ui-onboarding": "^1.2.0-beta.6",
+    "@apollosproject/ui-passes": "^1.2.0-beta.6",
+    "@apollosproject/ui-scripture": "^1.2.0-beta.6",
+    "@apollosproject/ui-storybook": "^1.2.0-beta.6",
     "@react-native-community/async-storage": "^1.5.1",
     "@react-native-community/geolocation": "^2.0.2",
     "apollo-cache-inmemory": "1.5.1",
diff --git a/apolloschurchapp/src/content-feed/content-feed.tests.js b/apolloschurchapp/src/content-feed/content-feed.tests.js
index 8578983..21b8c9c 100644
--- a/apolloschurchapp/src/content-feed/content-feed.tests.js
+++ b/apolloschurchapp/src/content-feed/content-feed.tests.js
@@ -38,7 +38,6 @@ describe('content feed query component', () => {
                       __typename: 'ContentChannel',
                     },
                     title: 'Mea Animal Aperiam Ornatus Eu',
-                    hyphenatedTitle: 'Mea Animal Aperiam Ornatus Eu',
                     summary: 'Bla bla bla',
                     theme: null,
                     isLiked: false,
@@ -60,8 +59,6 @@ describe('content feed query component', () => {
                       __typename: 'ContentChannel',
                     },
                     title: 'Probo Senserit Id Mea, Ut Sed Malis Postea,',
-                    hyphenatedTitle:
-                      'Probo Senserit Id Mea, Ut Sed Malis Postea,',
                     summary: 'Bla bla bla',
                     theme: null,
                     isLiked: false,
diff --git a/apolloschurchapp/src/content-single/DevotionalContentItem/index.js b/apolloschurchapp/src/content-single/DevotionalContentItem/index.js
index 07b4938..19b8898 100644
--- a/apolloschurchapp/src/content-single/DevotionalContentItem/index.js
+++ b/apolloschurchapp/src/content-single/DevotionalContentItem/index.js
@@ -2,6 +2,7 @@ import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
 import { SafeAreaView } from 'react-navigation';
 import { Query } from 'react-apollo';
+import { PagerScroll } from 'react-native-tab-view';
 
 import {
   ErrorCard,
@@ -98,6 +99,7 @@ class DevotionalContentItem extends PureComponent {
     return (
       <TabView
         routes={tabRoutes}
+        renderPager={(props) => <PagerScroll {...props} />}
         renderScene={SceneMap({
           content: this.contentRoute({ scriptures, loading }),
           scripture: this.scriptureRoute({ scriptures, loading }),
diff --git a/apolloschurchapp/src/tabs/connect/RecentlyLikedTileFeed/RecentlyLikedTileFeedConnected.tests.js b/apolloschurchapp/src/tabs/connect/RecentlyLikedTileFeed/RecentlyLikedTileFeedConnected.tests.js
index 016b246..a8620c1 100644
--- a/apolloschurchapp/src/tabs/connect/RecentlyLikedTileFeed/RecentlyLikedTileFeedConnected.tests.js
+++ b/apolloschurchapp/src/tabs/connect/RecentlyLikedTileFeed/RecentlyLikedTileFeedConnected.tests.js
@@ -36,7 +36,6 @@ describe('RecentlyLikedTileFeedConnected', () => {
                     __typename: 'ContentChannel',
                   },
                   title: 'Mea Animal Aperiam Ornatus Eu',
-                  hyphenatedTitle: 'Mea Animal Aperiam Ornatus Eu',
                   summary: 'Bla bla bla',
                   theme: null,
                   isLiked: false,
@@ -58,8 +57,6 @@ describe('RecentlyLikedTileFeedConnected', () => {
                     __typename: 'ContentChannel',
                   },
                   title: 'Probo Senserit Id Mea, Ut Sed Malis Postea,',
-                  hyphenatedTitle:
-                    'Probo Senserit Id Mea, Ut Sed Malis Postea,',
                   summary: 'Bla bla bla',
                   theme: null,
                   isLiked: false,
diff --git a/apolloschurchapp/src/tabs/discover/DiscoverFeed/DiscoverFeed.tests.js b/apolloschurchapp/src/tabs/discover/DiscoverFeed/DiscoverFeed.tests.js
index 568ee71..3ce5dec 100644
--- a/apolloschurchapp/src/tabs/discover/DiscoverFeed/DiscoverFeed.tests.js
+++ b/apolloschurchapp/src/tabs/discover/DiscoverFeed/DiscoverFeed.tests.js
@@ -29,7 +29,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:00bb7b364911281c97fc50f2a0d17b11',
                       title: 'Anderson Family Cookout!',
-                      hyphenatedTitle: 'Anderson Family Cookout!',
                       isLiked: false,
                       likedCount: 0,
                       summary: 'Come join us!',
@@ -61,8 +60,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:7db601b238aa4345a818654300f83dd2',
                       title: 'Robert Madu is Coming to Christ Fellowship',
-                      hyphenatedTitle:
-                        'Robert Madu is Coming to Christ Fellowship',
                       isLiked: false,
                       likedCount: 5,
                       summary:
@@ -95,7 +92,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:296373ecb53580855cadffa0375ebe18',
                       title: 'Guys Night!',
-                      hyphenatedTitle: 'Guys Night!',
                       isLiked: false,
                       likedCount: 0,
                       summary:
@@ -138,7 +134,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:604a4a763a5fc0edfe386684ca6b3515',
                       title: 'Test Article',
-                      hyphenatedTitle: 'Test Article',
                       isLiked: false,
                       likedCount: 0,
                       summary: 'Test article...',
@@ -160,8 +155,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:c495ff18cd998ed516a798b6218907cd',
                       title: '10 ways to refresh your spirit every day',
-                      hyphenatedTitle:
-                        '10 ways to refresh your spirit every day',
                       isLiked: false,
                       likedCount: 2,
                       summary:
@@ -184,7 +177,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
                       title: 'Live for Freedom Toolkit',
-                      hyphenatedTitle: 'Live for Freedom Toolkit',
                       isLiked: false,
                       likedCount: 1,
                       summary:
@@ -217,7 +209,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'ContentSeriesContentItem:6d8dc8e4bad017f815ae0cac8bf692bb',
                       title: 'The New Weekend Experience',
-                      hyphenatedTitle: 'The New Weekend Experience',
                       isLiked: false,
                       likedCount: 1,
                       summary:
@@ -253,7 +244,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'ContentSeriesContentItem:08449058d438ebeaffe9adb7a8e633cc',
                       title: 'Gathering',
-                      hyphenatedTitle: 'Gathering',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -288,7 +278,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'ContentSeriesContentItem:b74c2d295a0cf1340232ffa8be856c00',
                       title: 'Gauntlet 2018',
-                      hyphenatedTitle: 'Gauntlet 2018',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -333,7 +322,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'ContentSeriesContentItem:b085ec27d53006e9e6b95f3ac57da581',
                       title: 'Jeremiah: A 33-Day Devotional',
-                      hyphenatedTitle: 'Jeremiah: A 33-Day Devotional',
                       isLiked: true,
                       likedCount: 6,
                       summary: 'Have you ever had to deliver difficult news?',
@@ -365,8 +353,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'ContentSeriesContentItem:71bfdac0b98ea1b72a63ff4ea64e3c5a',
                       title: 'How To Be A Better Leader: A 7-Day Devotional',
-                      hyphenatedTitle:
-                        'How To Be A Better Leader: A 7-Day Devotional',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -398,7 +384,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'ContentSeriesContentItem:44fd3b82373ea8f197bd1e1e77c4bed0',
                       title: '2 Thessalonians: A 5-Day Devotional',
-                      hyphenatedTitle: '2 Thessalonians: A 5-Day Devotional',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -441,8 +426,6 @@ describe('The DiscoverFeed component', () => {
                         'UniversalContentItem:cf8fa657f23af88a2d63f7e4a2237db8',
                       title:
                         'The surprising journey that changed me and my dad',
-                      hyphenatedTitle:
-                        'The surprising journey that changed me and my dad',
                       isLiked: false,
                       likedCount: 6,
                       summary:
@@ -475,8 +458,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:2e17f5a66407d114ab9f2392b03ebccd',
                       title: 'When I lost hope, my church family had my back',
-                      hyphenatedTitle:
-                        'When I lost hope, my church family had my back',
                       isLiked: true,
                       likedCount: 5,
                       summary:
@@ -509,7 +490,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:ae8ec75906ba7437c49ad2534b5024db',
                       title: 'A Place to Worship Free of Fear',
-                      hyphenatedTitle: 'A Place to Worship Free of Fear',
                       isLiked: false,
                       likedCount: 1,
                       summary:
diff --git a/apolloschurchapp/src/tabs/discover/SearchFeed/SearchFeed.tests.js b/apolloschurchapp/src/tabs/discover/SearchFeed/SearchFeed.tests.js
index aed2839..6dbc4da 100644
--- a/apolloschurchapp/src/tabs/discover/SearchFeed/SearchFeed.tests.js
+++ b/apolloschurchapp/src/tabs/discover/SearchFeed/SearchFeed.tests.js
@@ -40,7 +40,6 @@ describe('The SearchFeed component', () => {
                 node: {
                   id: 'DevotionalContentItem:561dfb7dbd8a5c093fd8385c7edaadbc',
                   title: 'How to lead people to Jesus',
-                  hyphenatedTitle: '-',
                   isLiked: false,
                   likedCount: 0,
                   summary:
@@ -77,7 +76,6 @@ describe('The SearchFeed component', () => {
                 node: {
                   id: 'UniversalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
                   title: 'Live for Freedom Toolkit',
-                  hyphenatedTitle: '-',
                   isLiked: false,
                   likedCount: 1,
                   summary:
@@ -114,7 +112,6 @@ describe('The SearchFeed component', () => {
                 node: {
                   id: 'UniversalContentItem:895738f70482712adb3ab45a08c30456',
                   title: 'Should your 20-something live at home?',
-                  hyphenatedTitle: '-',
                   isLiked: false,
                   likedCount: 0,
                   summary:
@@ -160,7 +157,6 @@ describe('The SearchFeed component', () => {
                 node: {
                   id: 'UniversalContentItem:296373ecb53580855cadffa0375ebe18',
                   title: 'Guys Night!',
-                  hyphenatedTitle: '-',
                   isLiked: false,
                   likedCount: 0,
                   summary: 'A time to connect with other guys who live nearby.',
@@ -196,7 +192,6 @@ describe('The SearchFeed component', () => {
                 node: {
                   id: 'UniversalContentItem:c495ff18cd998ed516a798b6218907cd',
                   title: '10 ways to refresh your spirit every day',
-                  hyphenatedTitle: '-',
                   isLiked: false,
                   likedCount: 2,
                   summary:
@@ -233,7 +228,6 @@ describe('The SearchFeed component', () => {
                 node: {
                   id: 'DevotionalContentItem:4dd74ceca9e09b5276e93cbf53821bd6',
                   title: 'Why leadership isn’t all about the title',
-                  hyphenatedTitle: '-',
                   isLiked: false,
                   likedCount: 0,
                   summary:
diff --git a/apolloschurchapp/src/tabs/discover/discover.tests.js b/apolloschurchapp/src/tabs/discover/discover.tests.js
index abd60dd..93fd091 100644
--- a/apolloschurchapp/src/tabs/discover/discover.tests.js
+++ b/apolloschurchapp/src/tabs/discover/discover.tests.js
@@ -29,7 +29,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:00bb7b364911281c97fc50f2a0d17b11',
                       title: 'Anderson Family Cookout!',
-                      hyphenatedTitle: 'Anderson Family Cookout!',
                       isLiked: false,
                       likedCount: 0,
                       summary: 'Come join us!',
@@ -61,8 +60,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:7db601b238aa4345a818654300f83dd2',
                       title: 'Robert Madu is Coming to Christ Fellowship',
-                      hyphenatedTitle:
-                        'Robert Madu is Coming to Christ Fellowship',
                       isLiked: false,
                       likedCount: 5,
                       summary:
@@ -95,7 +92,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:296373ecb53580855cadffa0375ebe18',
                       title: 'Guys Night!',
-                      hyphenatedTitle: 'Guys Night!',
                       isLiked: false,
                       likedCount: 0,
                       summary:
@@ -138,7 +134,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:604a4a763a5fc0edfe386684ca6b3515',
                       title: 'Test Article',
-                      hyphenatedTitle: 'Test Article',
                       isLiked: false,
                       likedCount: 0,
                       summary: 'Test article...',
@@ -160,8 +155,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:c495ff18cd998ed516a798b6218907cd',
                       title: '10 ways to refresh your spirit every day',
-                      hyphenatedTitle:
-                        '10 ways to refresh your spirit every day',
                       isLiked: false,
                       likedCount: 2,
                       summary:
@@ -184,7 +177,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
                       title: 'Live for Freedom Toolkit',
-                      hyphenatedTitle: 'Live for Freedom Toolkit',
                       isLiked: false,
                       likedCount: 1,
                       summary:
@@ -217,7 +209,6 @@ describe('The Discover tab component', () => {
                       id:
                         'ContentSeriesContentItem:6d8dc8e4bad017f815ae0cac8bf692bb',
                       title: 'The New Weekend Experience',
-                      hyphenatedTitle: 'The New Weekend Experience',
                       isLiked: false,
                       likedCount: 1,
                       summary:
@@ -253,7 +244,6 @@ describe('The Discover tab component', () => {
                       id:
                         'ContentSeriesContentItem:08449058d438ebeaffe9adb7a8e633cc',
                       title: 'Gathering',
-                      hyphenatedTitle: 'Gathering',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -288,7 +278,6 @@ describe('The Discover tab component', () => {
                       id:
                         'ContentSeriesContentItem:b74c2d295a0cf1340232ffa8be856c00',
                       title: 'Gauntlet 2018',
-                      hyphenatedTitle: 'Gauntlet 2018',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -333,7 +322,6 @@ describe('The Discover tab component', () => {
                       id:
                         'ContentSeriesContentItem:b085ec27d53006e9e6b95f3ac57da581',
                       title: 'Jeremiah: A 33-Day Devotional',
-                      hyphenatedTitle: 'Jeremiah: A 33-Day Devotional',
                       isLiked: true,
                       likedCount: 6,
                       summary: 'Have you ever had to deliver difficult news?',
@@ -365,8 +353,6 @@ describe('The Discover tab component', () => {
                       id:
                         'ContentSeriesContentItem:71bfdac0b98ea1b72a63ff4ea64e3c5a',
                       title: 'How To Be A Better Leader: A 7-Day Devotional',
-                      hyphenatedTitle:
-                        'How To Be A Better Leader: A 7-Day Devotional',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -398,7 +384,6 @@ describe('The Discover tab component', () => {
                       id:
                         'ContentSeriesContentItem:44fd3b82373ea8f197bd1e1e77c4bed0',
                       title: '2 Thessalonians: A 5-Day Devotional',
-                      hyphenatedTitle: '2 Thessalonians: A 5-Day Devotional',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -441,8 +426,6 @@ describe('The Discover tab component', () => {
                         'UniversalContentItem:cf8fa657f23af88a2d63f7e4a2237db8',
                       title:
                         'The surprising journey that changed me and my dad',
-                      hyphenatedTitle:
-                        'The surprising journey that changed me and my dad',
                       isLiked: false,
                       likedCount: 6,
                       summary:
@@ -475,8 +458,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:2e17f5a66407d114ab9f2392b03ebccd',
                       title: 'When I lost hope, my church family had my back',
-                      hyphenatedTitle:
-                        'When I lost hope, my church family had my back',
                       isLiked: true,
                       likedCount: 5,
                       summary:
@@ -509,7 +490,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:ae8ec75906ba7437c49ad2534b5024db',
                       title: 'A Place to Worship Free of Fear',
-                      hyphenatedTitle: 'A Place to Worship Free of Fear',
                       isLiked: false,
                       likedCount: 1,
                       summary:
diff --git a/apolloschurchapp/src/ui/ContentCardConnected/query.js b/apolloschurchapp/src/ui/ContentCardConnected/query.js
index 8b7d514..1ff178a 100644
--- a/apolloschurchapp/src/ui/ContentCardConnected/query.js
+++ b/apolloschurchapp/src/ui/ContentCardConnected/query.js
@@ -31,7 +31,6 @@ export const BASE_CARD_FRAGMENT = gql`
     ...coverImageFragment
     ...themeFragment
     title
-    hyphenatedTitle: title(hyphenated: true)
     summary
     isLiked
     ... on MediaContentItem {
diff --git a/apolloschurchapp/src/ui/HorizontalContentCardConnected/horizontalContentCardComponentMapper.js b/apolloschurchapp/src/ui/HorizontalContentCardConnected/horizontalContentCardComponentMapper.js
index 433b143..c658e47 100644
--- a/apolloschurchapp/src/ui/HorizontalContentCardConnected/horizontalContentCardComponentMapper.js
+++ b/apolloschurchapp/src/ui/HorizontalContentCardConnected/horizontalContentCardComponentMapper.js
@@ -1,32 +1,22 @@
 import React from 'react';
 import { get } from 'lodash';
-import PropTypes from 'prop-types';
 
 import {
   HorizontalDefaultCard,
   HorizontalHighlightCard,
 } from '@apollosproject/ui-kit';
 
-const horizontalContentCardComponentMapper = ({
-  title,
-  hyphenatedTitle,
-  ...props
-}) => {
+const horizontalContentCardComponentMapper = (props) => {
   // map typename to the the card we want to render.
   switch (get(props, '__typename')) {
     case 'MediaContentItem':
     case 'WeekendContentItem':
     case 'ContentSeriesContentItem':
     case 'DevotionalContentItem':
-      return <HorizontalHighlightCard title={hyphenatedTitle} {...props} />;
+      return <HorizontalHighlightCard {...props} />;
     default:
-      return <HorizontalDefaultCard title={title} {...props} />;
+      return <HorizontalDefaultCard {...props} />;
   }
 };
 
-horizontalContentCardComponentMapper.propTypes = {
-  hyphenatedTitle: PropTypes.string,
-  title: PropTypes.string,
-};
-
 export default horizontalContentCardComponentMapper;
diff --git a/apolloschurchapp/src/user-settings/Locations/Locations.tests.js b/apolloschurchapp/src/user-settings/Locations/Locations.tests.js
index 25ebf12..9e131ae 100644
--- a/apolloschurchapp/src/user-settings/Locations/Locations.tests.js
+++ b/apolloschurchapp/src/user-settings/Locations/Locations.tests.js
@@ -110,15 +110,6 @@ describe('Display Native Map with Locations', () => {
       result: {
         data: {
           campuses,
-          currentUser: {
-            id: 'AuthenticatedUser:123',
-            __typename: 'AuthenticatedUser',
-            profile: {
-              id: 'Person:123',
-              __typename: 'Person',
-              campus: campuses[3],
-            },
-          },
         },
       },
     };
diff --git a/apolloschurchapp/src/user-settings/Locations/MapView.js b/apolloschurchapp/src/user-settings/Locations/MapView.js
index 7f9b2bc..9478cf7 100644
--- a/apolloschurchapp/src/user-settings/Locations/MapView.js
+++ b/apolloschurchapp/src/user-settings/Locations/MapView.js
@@ -7,7 +7,6 @@ import { debounce } from 'lodash';
 
 import {
   Button,
-  Touchable,
   PaddedView,
   FlexedView,
   styled,
@@ -48,10 +47,6 @@ class MapView extends Component {
         longitude: PropTypes.number.isRequired,
       })
     ),
-    currentCampus: PropTypes.shape({
-      latitude: PropTypes.number.isRequired,
-      longitude: PropTypes.number.isRequired,
-    }),
     onLocationSelect: PropTypes.func.isRequired,
     initialRegion: PropTypes.shape({
       latitude: PropTypes.number.isRequired,
@@ -75,8 +70,6 @@ class MapView extends Component {
 
   animation = new Animated.Value(0);
 
-  scrollView = null;
-
   componentDidMount() {
     this.animation.addListener(debounce(this.updateCoordinates));
   }
@@ -91,33 +84,13 @@ class MapView extends Component {
     const cardIndex = Math.floor(
       this.previousScrollPosition / CARD_WIDTH + 0.3
     ); // animate 30% away from landing on the next item;
-    return this.sortedCampuses[cardIndex];
-  }
-
-  get sortedCampuses() {
-    const { currentCampus = null, campuses = [] } = this.props;
-    if (!this.props.currentCampus) {
-      return campuses;
-    }
-    return [
-      currentCampus,
-      ...campuses.filter(({ id }) => id !== currentCampus.id),
-    ];
+    return this.props.campuses[cardIndex];
   }
 
-  scrollToIndex = (index) => {
-    this.scrollView.getNode().scrollTo({
-      x: index * (CARD_WIDTH + 8),
-      y: 0,
-      animated: true,
-    });
-    this.updateCoordinates({ value: index * (CARD_WIDTH + 8) });
-  };
-
   updateCoordinates = ({ value }) => {
     this.previousScrollPosition = value;
 
-    const { userLocation } = this.props;
+    const { userLocation, campuses } = this.props;
     // campus card height + some padding
 
     const bottomPadding = 100 + this.props.theme.sizing.baseUnit * 12;
@@ -135,7 +108,7 @@ class MapView extends Component {
 
     const visibleCampuses = [
       userLocation,
-      ...(this.currentCampus ? [this.currentCampus] : this.sortedCampuses),
+      ...(this.currentCampus ? [this.currentCampus] : campuses),
     ];
 
     this.map.fitToCoordinates(visibleCampuses, {
@@ -144,8 +117,8 @@ class MapView extends Component {
   };
 
   render() {
-    const { onLocationSelect } = this.props;
-    const interpolations = this.sortedCampuses.map((marker, index) => {
+    const { campuses = [], onLocationSelect } = this.props;
+    const interpolations = campuses.map((marker, index) => {
       const inputRange = [
         (index - 1) * CARD_WIDTH,
         index * CARD_WIDTH,
@@ -168,13 +141,12 @@ class MapView extends Component {
             this.map = map;
           }}
         >
-          {this.sortedCampuses.map((campus, index) => {
+          {campuses.map((campus, index) => {
             const campusOpacity = {
               opacity: interpolations[index].opacity,
             };
             return (
               <Marker
-                onPress={() => this.scrollToIndex(index)}
                 key={campus.id}
                 opacityStyle={campusOpacity}
                 latitude={campus.latitude}
@@ -193,7 +165,6 @@ class MapView extends Component {
             contentContainerStyle={{
               paddingHorizontal: this.props.theme.sizing.baseUnit * 0.75,
             }}
-            ref={(ref) => (this.scrollView = ref)} // eslint-disable-line
             scrollEventThrottle={16} // roughtly 1000ms/60fps = 16ms
             onScroll={Animated.event(
               [
@@ -208,18 +179,14 @@ class MapView extends Component {
               { useNativeDriver: true }
             )}
           >
-            {this.sortedCampuses.map((campus) => (
-              <Touchable
+            {campuses.map((campus) => (
+              <StyledCampusCard
                 key={campus.id}
-                onPress={() => onLocationSelect(campus)}
-              >
-                <StyledCampusCard
-                  distance={campus.distanceFromLocation}
-                  title={campus.name}
-                  description={getCampusAddress(campus)}
-                  images={[campus.image]}
-                />
-              </Touchable>
+                distance={campus.distanceFromLocation}
+                title={campus.name}
+                description={getCampusAddress(campus)}
+                images={[campus.image]}
+              />
             ))}
           </Animated.ScrollView>
           <PaddedView>
@@ -228,7 +195,7 @@ class MapView extends Component {
               pill={false}
               type="secondary"
               onPress={() =>
-                onLocationSelect(this.currentCampus || this.sortedCampuses[0])
+                onLocationSelect(this.currentCampus || campuses[0])
               }
             />
           </PaddedView>
diff --git a/apolloschurchapp/src/user-settings/Locations/Marker.js b/apolloschurchapp/src/user-settings/Locations/Marker.js
index b91f89e..0c98fa9 100644
--- a/apolloschurchapp/src/user-settings/Locations/Marker.js
+++ b/apolloschurchapp/src/user-settings/Locations/Marker.js
@@ -25,8 +25,8 @@ const MarkerRingView = styled(({ theme }) => ({
   justifyContent: 'center',
 }))(View);
 
-const StyledMarker = memo(({ latitude, longitude, opacityStyle, onPress }) => (
-  <Marker onPress={onPress} coordinate={{ latitude, longitude }}>
+const StyledMarker = memo(({ latitude, longitude, opacityStyle }) => (
+  <Marker coordinate={{ latitude, longitude }}>
     <Animated.View style={opacityStyle}>
       <MarkerRingView>
         <MarkerView />
@@ -40,7 +40,6 @@ StyledMarker.displayName = 'StyledMarker';
 StyledMarker.propTypes = {
   latitude: PropTypes.number.isRequired,
   longitude: PropTypes.number.isRequired,
-  onPress: PropTypes.func.isRequired,
   opacityStyle: PropTypes.shape({}),
 };
 
diff --git a/apolloschurchapp/src/user-settings/Locations/getCampusLocations.js b/apolloschurchapp/src/user-settings/Locations/getCampusLocations.js
index 6dc73e9..1c98bb8 100644
--- a/apolloschurchapp/src/user-settings/Locations/getCampusLocations.js
+++ b/apolloschurchapp/src/user-settings/Locations/getCampusLocations.js
@@ -6,15 +6,6 @@ export default gql`
     campuses(location: { latitude: $latitude, longitude: $longitude }) {
       ...CampusParts
     }
-    currentUser {
-      id
-      profile {
-        id
-        campus {
-          ...CampusParts
-        }
-      }
-    }
   }
   ${CampusFragment}
 `;
diff --git a/apolloschurchapp/src/user-settings/Locations/index.js b/apolloschurchapp/src/user-settings/Locations/index.js
index d54cc06..c730500 100644
--- a/apolloschurchapp/src/user-settings/Locations/index.js
+++ b/apolloschurchapp/src/user-settings/Locations/index.js
@@ -4,7 +4,6 @@ import { Query, Mutation } from 'react-apollo';
 import { Dimensions } from 'react-native';
 import Geolocation from 'react-native-geolocation-service';
 import { PaddedView, ButtonLink } from '@apollosproject/ui-kit';
-import { get } from 'lodash';
 
 import GET_CAMPUSES from './getCampusLocations';
 import CHANGE_CAMPUS from './campusChange';
@@ -79,7 +78,7 @@ class Location extends PureComponent {
         }}
         fetchPolicy="cache-and-network"
       >
-        {({ loading, error, data: { campuses = [], currentUser } = {} }) => (
+        {({ loading, error, data: { campuses = [] } = {} }) => (
           <Mutation mutation={CHANGE_CAMPUS}>
             {(handlePress) => (
               <MapView
@@ -89,7 +88,6 @@ class Location extends PureComponent {
                 campuses={campuses}
                 initialRegion={this.props.initialRegion}
                 userLocation={this.state.userLocation}
-                currentCampus={get(currentUser, 'profile.campus')}
                 onLocationSelect={async ({ id }) => {
                   await handlePress({
                     variables: {
diff --git a/apolloschurchapp/src/user-settings/index.js b/apolloschurchapp/src/user-settings/index.js
index 3dfc18c..491b707 100644
--- a/apolloschurchapp/src/user-settings/index.js
+++ b/apolloschurchapp/src/user-settings/index.js
@@ -3,7 +3,6 @@ import { ScrollView } from 'react-native';
 import { StackActions, NavigationActions } from 'react-navigation';
 import PropTypes from 'prop-types';
 import { Query, Mutation } from 'react-apollo';
-import { getVersion, getBuildNumber } from 'react-native-device-info';
 
 import {
   BackgroundView,
@@ -152,13 +151,6 @@ class UserSettings extends PureComponent {
                           )}
                         </Mutation>
                       </TableView>
-                      <TableView>
-                        <Cell>
-                          <CellText>
-                            {`App Version: ${getVersion()}.${getBuildNumber()}`}
-                          </CellText>
-                        </Cell>
-                      </TableView>
                     </>
                   )}
                 </WebBrowserConsumer>
diff --git a/apolloschurchapp/src/user-settings/uploadPhoto.js b/apolloschurchapp/src/user-settings/uploadPhoto.js
index c1cc667..7eb5815 100644
--- a/apolloschurchapp/src/user-settings/uploadPhoto.js
+++ b/apolloschurchapp/src/user-settings/uploadPhoto.js
@@ -9,8 +9,6 @@ const options = {
   storageOptions: {
     skipBackup: true,
     path: 'images',
-    waitUntilSaved: true,
-    cameraRoll: true,
   },
 };
 
