diff --git a/apolloschurchapp/android/app/build.gradle b/apolloschurchapp/android/app/build.gradle
index 06ba626..2ff1813 100644
--- a/apolloschurchapp/android/app/build.gradle
+++ b/apolloschurchapp/android/app/build.gradle
@@ -146,7 +146,7 @@ android {
         applicationId "com.apolloschurch.app"
         minSdkVersion rootProject.ext.minSdkVersion
         targetSdkVersion rootProject.ext.targetSdkVersion
-        versionCode 24
+        versionCode 20
         versionName "1.2.0"
         ndk {
             abiFilters "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
diff --git a/apolloschurchapp/apollos.json b/apolloschurchapp/apollos.json
index 38517b2..6a4e273 100644
--- a/apolloschurchapp/apollos.json
+++ b/apolloschurchapp/apollos.json
@@ -1,4 +1,4 @@
 {
-    "version": "1.2.0-beta.7",
+    "version": "1.2.0-beta.0",
     "environment": "client"
 }
\ No newline at end of file
diff --git a/apolloschurchapp/ios/Podfile.lock b/apolloschurchapp/ios/Podfile.lock
index a30f91b..8bf013d 100644
--- a/apolloschurchapp/ios/Podfile.lock
+++ b/apolloschurchapp/ios/Podfile.lock
@@ -1,6 +1,6 @@
 PODS:
   - boost-for-react-native (1.63.0)
-  - BVLinearGradient (2.5.4):
+  - BVLinearGradient (2.5.6):
     - React
   - DoubleConversion (1.1.6)
   - Folly (2018.10.22.00):
@@ -76,16 +76,14 @@ PODS:
     - React
   - react-native-safari-view (1.0.0):
     - React
-  - react-native-safe-area-context (0.3.6):
-    - React
   - react-native-splash-screen (3.2.0):
     - React
-  - react-native-video (5.0.0):
+  - react-native-video (5.0.2):
     - React
-    - react-native-video/Video (= 5.0.0)
-  - react-native-video/Video (5.0.0):
+    - react-native-video/Video (= 5.0.2)
+  - react-native-video/Video (5.0.2):
     - React
-  - react-native-webview (5.10.0):
+  - react-native-webview (5.12.1):
     - React
   - React-RCTActionSheet (0.60.5):
     - React-Core (= 0.60.5)
@@ -114,17 +112,17 @@ PODS:
     - React-Core
   - RNAirplay (1.0.0):
     - React
-  - RNCAsyncStorage (1.5.1):
+  - RNCAsyncStorage (1.6.1):
     - React
-  - RNDeviceInfo (4.0.1):
+  - RNDeviceInfo (2.3.2):
     - React
-  - RNGestureHandler (1.3.0):
+  - RNGestureHandler (1.4.1):
     - React
   - RNInAppBrowser (3.1.0):
     - React
   - RNScreens (1.0.0-alpha.23):
     - React
-  - RNSVG (9.4.0):
+  - RNSVG (9.9.3):
     - React
   - yoga (0.60.5.React)
 
@@ -148,7 +146,6 @@ DEPENDENCIES:
   - react-native-onesignal (from `../node_modules/react-native-onesignal`)
   - react-native-passkit-wallet (from `../node_modules/react-native-passkit-wallet`)
   - react-native-safari-view (from `../node_modules/react-native-safari-view`)
-  - react-native-safe-area-context (from `../node_modules/react-native-safe-area-context`)
   - react-native-splash-screen (from `../node_modules/react-native-splash-screen`)
   - react-native-video (from `../node_modules/react-native-video`)
   - react-native-webview (from `../node_modules/react-native-webview`)
@@ -216,8 +213,6 @@ EXTERNAL SOURCES:
     :path: "../node_modules/react-native-passkit-wallet"
   react-native-safari-view:
     :path: "../node_modules/react-native-safari-view"
-  react-native-safe-area-context:
-    :path: "../node_modules/react-native-safe-area-context"
   react-native-splash-screen:
     :path: "../node_modules/react-native-splash-screen"
   react-native-video:
@@ -265,7 +260,7 @@ EXTERNAL SOURCES:
 
 SPEC CHECKSUMS:
   boost-for-react-native: 39c7adb57c4e60d6c5479dd8623128eb5b3f0f2c
-  BVLinearGradient: 8cbc5155c978f2e43098818c91d206d07aae6d30
+  BVLinearGradient: e3aad03778a456d77928f594a649e96995f1c872
   DoubleConversion: 5805e889d232975c086db112ece9ed034df7a0b2
   Folly: 30e7936e1c45c08d884aa59369ed951a8e68cf51
   glog: 1f3da668190260b06b429bb211bfbee5cd790c28
@@ -285,10 +280,9 @@ SPEC CHECKSUMS:
   react-native-onesignal: 8958311abbd9125f26e600914e48833b57321cbf
   react-native-passkit-wallet: 51e3babdaeec4c10b5736786ebd1bfb470fcaa36
   react-native-safari-view: 955d7160d159241b8e9395d12d10ea0ef863dcdd
-  react-native-safe-area-context: e380a6f783ccaec848e2f3cc8eb205a62362950d
   react-native-splash-screen: 200d11d188e2e78cea3ad319964f6142b6384865
-  react-native-video: 6555881252c8ca039760e1cd6df28ac28ffb2baf
-  react-native-webview: 3c79685b61a68d1fe99136f98f2f34b04dce2e02
+  react-native-video: d01ed7ff1e38fa7dcc6c15c94cf505e661b7bfd0
+  react-native-webview: 5036ed4a973ad1deeeff118262d2df7b60b3419d
   React-RCTActionSheet: b0f1ea83f4bf75fb966eae9bfc47b78c8d3efd90
   React-RCTAnimation: 359ba1b5690b1e87cc173558a78e82d35919333e
   React-RCTBlob: 5e2b55f76e9a1c7ae52b826923502ddc3238df24
@@ -301,12 +295,12 @@ SPEC CHECKSUMS:
   React-RCTWebSocket: cd932a16b7214898b6b7f788c8bddb3637246ac4
   rn-fetch-blob: 651b8d076b43d0d7aa294a3d9ec16c00aab8bef9
   RNAirplay: 75777dd55dafbd3687703c3eb9c290ea6cc43248
-  RNCAsyncStorage: b63d6e83fc629b01df6b624688f17944cea5637f
-  RNDeviceInfo: 687c1b2ab6d86ff1ca1208783320cd144138c7f2
-  RNGestureHandler: 5329a942fce3d41c68b84c2c2276ce06a696d8b0
+  RNCAsyncStorage: 2e2e3feb9bdadc752a026703d8c4065ca912e75a
+  RNDeviceInfo: 17e34f6dd902f08d88cbe2c0b7a01be948d43641
+  RNGestureHandler: 4cb47a93019c1a201df2644413a0a1569a51c8aa
   RNInAppBrowser: 9d1ffb9db0d0cbdf03f528f1f0624b2c533b93fd
   RNScreens: f28b48b8345f2f5f39ed6195518291515032a788
-  RNSVG: 9cb6e958c4b6a1f58185ac72a350b148947d6fed
+  RNSVG: 2a15d0d14d171ba610db927190604d1ab223f413
   yoga: 312528f5bbbba37b4dcea5ef00e8b4033fdd9411
 
 PODFILE CHECKSUM: d235853c2d054909a0114df39a2529360918990d
diff --git a/apolloschurchapp/ios/apolloschurchapp-tvOS/Info.plist b/apolloschurchapp/ios/apolloschurchapp-tvOS/Info.plist
index 52fd423..8b8dba9 100644
--- a/apolloschurchapp/ios/apolloschurchapp-tvOS/Info.plist
+++ b/apolloschurchapp/ios/apolloschurchapp-tvOS/Info.plist
@@ -19,7 +19,7 @@
 	<key>CFBundleSignature</key>
 	<string>????</string>
 	<key>CFBundleVersion</key>
-	<string>24</string>
+	<string>20</string>
 	<key>LSRequiresIPhoneOS</key>
 	<true />
 	<key>UILaunchStoryboardName</key>
diff --git a/apolloschurchapp/ios/apolloschurchapp-tvOSTests/Info.plist b/apolloschurchapp/ios/apolloschurchapp-tvOSTests/Info.plist
index 7188b0b..261aeea 100644
--- a/apolloschurchapp/ios/apolloschurchapp-tvOSTests/Info.plist
+++ b/apolloschurchapp/ios/apolloschurchapp-tvOSTests/Info.plist
@@ -19,6 +19,6 @@
 	<key>CFBundleSignature</key>
 	<string>????</string>
 	<key>CFBundleVersion</key>
-	<string>24</string>
+	<string>20</string>
 </dict>
 </plist>
diff --git a/apolloschurchapp/ios/apolloschurchapp.xcodeproj/project.pbxproj b/apolloschurchapp/ios/apolloschurchapp.xcodeproj/project.pbxproj
index a24c843..b75c22e 100644
--- a/apolloschurchapp/ios/apolloschurchapp.xcodeproj/project.pbxproj
+++ b/apolloschurchapp/ios/apolloschurchapp.xcodeproj/project.pbxproj
@@ -672,7 +672,7 @@
 			baseConfigurationReference = 54F271B8EB67A4997A2C7D96 /* Pods-apolloschurchapp.debug.xcconfig */;
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
-				CURRENT_PROJECT_VERSION = 18;
+				CURRENT_PROJECT_VERSION = 16;
 				DEAD_CODE_STRIPPING = NO;
 				DEVELOPMENT_TEAM = 7TM8C6252Z;
 				HEADER_SEARCH_PATHS = (
@@ -698,7 +698,7 @@
 			baseConfigurationReference = 1F8F5BC4745C8F22C695573D /* Pods-apolloschurchapp.release.xcconfig */;
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
-				CURRENT_PROJECT_VERSION = 18;
+				CURRENT_PROJECT_VERSION = 16;
 				HEADER_SEARCH_PATHS = (
 					"$(inherited)",
 					"$(SRCROOT)/../node_modules/react-native-video/ios",
diff --git a/apolloschurchapp/ios/apolloschurchapp/Info.plist b/apolloschurchapp/ios/apolloschurchapp/Info.plist
index 4cbb689..7f789f7 100644
--- a/apolloschurchapp/ios/apolloschurchapp/Info.plist
+++ b/apolloschurchapp/ios/apolloschurchapp/Info.plist
@@ -30,7 +30,7 @@
 		</dict>
 	</array>
 	<key>CFBundleVersion</key>
-	<string>25</string>
+	<string>21</string>
 	<key>ITSAppUsesNonExemptEncryption</key>
 	<false />
 	<key>LSRequiresIPhoneOS</key>
@@ -55,10 +55,8 @@
 	</dict>
 	<key>NSCameraUsageDescription</key>
 	<string>$(PRODUCT_NAME) would like to use your camera (for uploading profile pictures)</string>
-	<key>NSLocationAlwaysUsageDescription</key>
-	<string>$(PRODUCT_NAME) would like to use your location for determining your closest campus</string>
 	<key>NSLocationWhenInUseUsageDescription</key>
-	<string>$(PRODUCT_NAME) would like to use your location for determining your closest campus</string>
+	<string />
 	<key>NSMicrophoneUsageDescription</key>
 	<string>$(PRODUCT_NAME) would like to your microphone (for videos)</string>
 	<key>NSPhotoLibraryAddUsageDescription</key>
diff --git a/apolloschurchapp/ios/apolloschurchappTests/Info.plist b/apolloschurchapp/ios/apolloschurchappTests/Info.plist
index 2892f83..dab023f 100644
--- a/apolloschurchapp/ios/apolloschurchappTests/Info.plist
+++ b/apolloschurchapp/ios/apolloschurchappTests/Info.plist
@@ -19,6 +19,6 @@
 	<key>CFBundleSignature</key>
 	<string>????</string>
 	<key>CFBundleVersion</key>
-	<string>24</string>
+	<string>20</string>
 </dict>
 </plist>
diff --git a/apolloschurchapp/jest.setup.js b/apolloschurchapp/jest.setup.js
index de5a421..946e5e0 100644
--- a/apolloschurchapp/jest.setup.js
+++ b/apolloschurchapp/jest.setup.js
@@ -1,16 +1,6 @@
 import React from 'react';
 import { NativeModules } from 'react-native';
 // We ran into an issue where SafeAreaView would break jest tests.
-
-jest.mock(
-  '../apollos-ui-kit/node_modules/react-native-safe-area-context/',
-  () => ({
-    SafeAreaConsumer: ({ children }) =>
-      children({ top: 0, bottom: 0, left: 0, right: 0 }),
-    SafeAreaProvider: ({ children }) => children,
-  })
-);
-
 jest.mock('react-navigation', () => {
   const ActualNavigation = require.requireActual('react-navigation');
   return {
@@ -59,7 +49,7 @@ jest.mock('react-native-safari-view', () => ({
 }));
 
 jest.mock('react-native-device-info', () => ({
-  getUniqueId: () => 'id-123',
+  getUniqueID: () => 'id-123',
   getSystemVersion: () => 'sys-version-123',
   getModel: () => 'ios',
   getVersion: () => 'version-123',
@@ -76,7 +66,6 @@ jest.mock('@apollosproject/ui-analytics', () => ({
   track: () => '',
   AnalyticsConsumer: ({ children }) => children({ test: jest.fn() }),
   AnalyticsProvider: ({ children }) => children,
-  TrackEventWhenLoaded: () => null,
   withTrackOnPress: (Component) => (props) => <Component {...props} />,
 }));
 
diff --git a/apolloschurchapp/metro.config.js b/apolloschurchapp/metro.config.js
index f833f5a..a5d875a 100644
--- a/apolloschurchapp/metro.config.js
+++ b/apolloschurchapp/metro.config.js
@@ -26,7 +26,6 @@ const localDeps = [
 
 const sharedNativeModules = [
   'react-native-linear-gradient',
-  'react-native-safe-area-context',
   'react-native-svg',
   'react-native-onesignal',
   'react-native-music-control',
diff --git a/apolloschurchapp/package.json b/apolloschurchapp/package.json
index 20d4fe7..90f80d2 100644
--- a/apolloschurchapp/package.json
+++ b/apolloschurchapp/package.json
@@ -1,6 +1,6 @@
 {
   "name": "apolloschurchapp",
-  "version": "1.2.0-beta.7",
+  "version": "1.2.0-beta.0",
   "private": true,
   "devDependencies": {
     "codecov": "3.0.2",
@@ -35,12 +35,12 @@
     "generate-stories": "rnstl --searchDir ./src --pattern **/*.stories.js",
     "lint": "eslint ./src --ext .js",
     "version": "node ../../scripts/update-apollos-version.js && git add apollos.json",
-    "postversion": "react-native-version --never-amend --never-increment-build --android android/app/build.gradle --ios ios/",
+    "postversion": "react-native-version --never-amend --android android/app/build.gradle --ios ios/",
     "prestart": "lerna run generate-stories & node ./scripts/get-introspection-data.js",
     "postinstall": "jetify",
     "pods": "./scripts/install_pods.sh",
     "start": "react-native start",
-    "test": "TZ=UTC jest"
+    "test": "jest"
   },
   "jest": {
     "preset": "react-native",
@@ -63,16 +63,16 @@
   },
   "dependencies": {
     "@apollosproject/react-native-airplay-btn": "0.2.0",
-    "@apollosproject/ui-analytics": "^1.2.0-beta.7",
-    "@apollosproject/ui-auth": "^1.2.0-beta.7",
-    "@apollosproject/ui-htmlview": "^1.2.0-beta.7",
-    "@apollosproject/ui-kit": "^1.2.0-beta.7",
-    "@apollosproject/ui-media-player": "^1.2.0-beta.7",
-    "@apollosproject/ui-notifications": "^1.2.0-beta.7",
-    "@apollosproject/ui-onboarding": "^1.2.0-beta.7",
-    "@apollosproject/ui-passes": "^1.2.0-beta.7",
-    "@apollosproject/ui-scripture": "^1.2.0-beta.7",
-    "@apollosproject/ui-storybook": "^1.2.0-beta.7",
+    "@apollosproject/ui-analytics": "^1.2.0-beta.0",
+    "@apollosproject/ui-auth": "^1.2.0-beta.0",
+    "@apollosproject/ui-htmlview": "^1.2.0-beta.0",
+    "@apollosproject/ui-kit": "^1.2.0-beta.0",
+    "@apollosproject/ui-media-player": "^1.2.0-beta.0",
+    "@apollosproject/ui-notifications": "^1.2.0-beta.0",
+    "@apollosproject/ui-onboarding": "^1.2.0-beta.0",
+    "@apollosproject/ui-passes": "^1.2.0-beta.0",
+    "@apollosproject/ui-scripture": "^1.2.0-beta.0",
+    "@apollosproject/ui-storybook": "^1.2.0-beta.0",
     "@react-native-community/async-storage": "^1.5.1",
     "@react-native-community/geolocation": "^2.0.2",
     "apollo-cache-inmemory": "1.5.1",
@@ -97,7 +97,7 @@
     "react-apollo": "2.5.5",
     "react-native": "0.60.5",
     "react-native-config": "^0.11.7",
-    "react-native-device-info": "^4.0.1",
+    "react-native-device-info": "^2.1.2",
     "react-native-geolocation-service": "^3.0.0",
     "react-native-gesture-handler": "^1.3.0",
     "react-native-image-picker": "0.28.0",
@@ -108,7 +108,6 @@
     "react-native-onesignal": "3.3.2",
     "react-native-passkit-wallet": "^0.1.4",
     "react-native-safari-view": "2.1.0",
-    "react-native-safe-area-context": "^0.3.6",
     "react-native-safe-area-view": "0.13.1",
     "react-native-screens": "^1.0.0-alpha.23",
     "react-native-splash-screen": "3.2.0",
diff --git a/apolloschurchapp/src/content-feed/content-feed.tests.js b/apolloschurchapp/src/content-feed/content-feed.tests.js
index 8578983..21b8c9c 100644
--- a/apolloschurchapp/src/content-feed/content-feed.tests.js
+++ b/apolloschurchapp/src/content-feed/content-feed.tests.js
@@ -38,7 +38,6 @@ describe('content feed query component', () => {
                       __typename: 'ContentChannel',
                     },
                     title: 'Mea Animal Aperiam Ornatus Eu',
-                    hyphenatedTitle: 'Mea Animal Aperiam Ornatus Eu',
                     summary: 'Bla bla bla',
                     theme: null,
                     isLiked: false,
@@ -60,8 +59,6 @@ describe('content feed query component', () => {
                       __typename: 'ContentChannel',
                     },
                     title: 'Probo Senserit Id Mea, Ut Sed Malis Postea,',
-                    hyphenatedTitle:
-                      'Probo Senserit Id Mea, Ut Sed Malis Postea,',
                     summary: 'Bla bla bla',
                     theme: null,
                     isLiked: false,
diff --git a/apolloschurchapp/src/content-single/HorizontalContentFeed/index.js b/apolloschurchapp/src/content-single/HorizontalContentFeed/index.js
index f5f356f..666d509 100644
--- a/apolloschurchapp/src/content-single/HorizontalContentFeed/index.js
+++ b/apolloschurchapp/src/content-single/HorizontalContentFeed/index.js
@@ -26,28 +26,18 @@ class HorizontalContentFeed extends Component {
     }),
   };
 
-  renderItem = ({ item }) => {
-    const itemId = get(item, 'id', '');
-    const disabled = get(item, 'id', '') === this.props.contentId;
-    return (
-      <TouchableScale
-        onPress={() => this.handleOnPressItem(item)}
-        disabled={disabled}
-      >
-        <HorizontalContentCardConnected
-          contentId={itemId}
-          disabled={disabled}
-        />
-      </TouchableScale>
-    );
-  };
-
   handleOnPressItem = (item) => {
     this.props.navigation.push('ContentSingle', {
       itemId: item.id,
     });
   };
 
+  renderItem = ({ item }) => (
+    <TouchableScale onPress={() => this.handleOnPressItem(item)}>
+      <HorizontalContentCardConnected contentId={get(item, 'id', '')} />
+    </TouchableScale>
+  );
+
   renderFeed = ({ data, loading, error, fetchMore }) => {
     if (error) return null;
     if (loading) return null;
diff --git a/apolloschurchapp/src/event/Event.js b/apolloschurchapp/src/event/Event.js
deleted file mode 100644
index 18a6d85..0000000
--- a/apolloschurchapp/src/event/Event.js
+++ /dev/null
@@ -1,68 +0,0 @@
-import React from 'react';
-import { get } from 'lodash';
-import PropTypes from 'prop-types';
-import moment from 'moment';
-import {
-  GradientOverlayImage,
-  BackgroundView,
-  PaddedView,
-  H2,
-  Paragraph,
-  StretchyView,
-} from '@apollosproject/ui-kit';
-import HTMLView from '@apollosproject/ui-htmlview';
-
-import { FlexedScrollView, EventInfoItem } from './components';
-
-const Event = ({ event, loading }) => {
-  const coverImageSources = get(event, 'image.sources', []);
-  return (
-    <BackgroundView>
-      <StretchyView>
-        {({ Stretchy, ...scrollViewProps }) => (
-          <FlexedScrollView {...scrollViewProps}>
-            {coverImageSources.length || loading ? (
-              <Stretchy>
-                <GradientOverlayImage
-                  isLoading={!coverImageSources.length && loading}
-                  source={coverImageSources}
-                />
-              </Stretchy>
-            ) : null}
-            <PaddedView vertical={false}>
-              <H2 padded isLoading={!event.name && loading}>
-                {event.name}
-              </H2>
-              <Paragraph>
-                <EventInfoItem
-                  icon={'time'}
-                  title={moment(event.start).format('ddd, MMMM Do, YYYY')}
-                  subtitle={`${moment(event.start).format('LT')} — ${moment(
-                    event.end
-                  ).format('LT')}`}
-                />
-                <EventInfoItem icon={'pin'} title={event.location} />
-              </Paragraph>
-              <HTMLView isLoading={loading}>{event.description}</HTMLView>
-            </PaddedView>
-          </FlexedScrollView>
-        )}
-      </StretchyView>
-    </BackgroundView>
-  );
-};
-
-Event.propTypes = {
-  event: PropTypes.shape({
-    __typename: PropTypes.string,
-    id: PropTypes.string,
-    name: PropTypes.string,
-    description: PropTypes.string,
-    start: PropTypes.string,
-    end: PropTypes.string,
-    location: PropTypes.string,
-  }),
-  loading: PropTypes.bool,
-};
-
-export default Event;
diff --git a/apolloschurchapp/src/event/Event.tests.js b/apolloschurchapp/src/event/Event.tests.js
deleted file mode 100644
index 10cf33d..0000000
--- a/apolloschurchapp/src/event/Event.tests.js
+++ /dev/null
@@ -1,31 +0,0 @@
-import React from 'react';
-import renderer from 'react-test-renderer';
-import moment from 'moment';
-
-import Providers from 'apolloschurchapp/src/Providers';
-
-import Event from './Event';
-
-describe('The Event component', () => {
-  it('should render', () => {
-    const tree = renderer.create(
-      <Providers>
-        <Event
-          event={{
-            __typename: 'Event',
-            id: 'Event:123',
-            start: moment('2019-09-26T15:10:51.200Z')
-              .utc()
-              .toJSON(),
-            end: moment('2019-09-26T17:10:51.200Z')
-              .utc()
-              .toJSON(),
-            location: 'Willow Creek, Chicago',
-            image: { sources: [{ url: 'https://url.com/image.jpg' }] },
-          }}
-        />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-});
diff --git a/apolloschurchapp/src/event/EventConnected.js b/apolloschurchapp/src/event/EventConnected.js
deleted file mode 100644
index 71f0a3b..0000000
--- a/apolloschurchapp/src/event/EventConnected.js
+++ /dev/null
@@ -1,81 +0,0 @@
-import React, { PureComponent } from 'react';
-import { Query } from 'react-apollo';
-import { get } from 'lodash';
-import PropTypes from 'prop-types';
-
-import { ErrorCard, ThemeMixin } from '@apollosproject/ui-kit';
-
-import { TrackEventWhenLoaded } from '@apollosproject/ui-analytics';
-
-import NavigationHeader from '../content-single/NavigationHeader';
-import Event from './Event';
-import GET_EVENT from './getEvent';
-
-class EventConnected extends PureComponent {
-  static propTypes = {
-    navigation: PropTypes.shape({
-      getParam: PropTypes.func,
-      push: PropTypes.func,
-    }),
-  };
-
-  static navigationOptions = {
-    header: NavigationHeader,
-    headerTransparent: true,
-    headerMode: 'float',
-  };
-
-  get eventId() {
-    return this.props.navigation.getParam('eventId', []);
-  }
-
-  get queryVariables() {
-    return { eventId: this.eventId };
-  }
-
-  renderWithData = ({ loading, error, data }) => {
-    if (error) return <ErrorCard error={error} />;
-
-    const event = data.node || {};
-
-    const { theme = {} } = event;
-
-    return (
-      <ThemeMixin
-        mixin={{
-          type: get(theme, 'type', 'light').toLowerCase(),
-          colors: get(theme, 'colors'),
-        }}
-      >
-        <TrackEventWhenLoaded
-          loaded={!!(!loading && event.name)}
-          eventName={'View Event'}
-          properties={{
-            title: event.name,
-            eventId: this.eventId,
-          }}
-        />
-        <Event
-          id={this.eventId}
-          event={event}
-          loading={loading}
-          error={error}
-        />
-      </ThemeMixin>
-    );
-  };
-
-  render() {
-    return (
-      <Query
-        query={GET_EVENT}
-        fetchPolicy="network-only"
-        variables={this.queryVariables}
-      >
-        {this.renderWithData}
-      </Query>
-    );
-  }
-}
-
-export default EventConnected;
diff --git a/apolloschurchapp/src/event/EventConnected.tests.js b/apolloschurchapp/src/event/EventConnected.tests.js
deleted file mode 100644
index 0594d24..0000000
--- a/apolloschurchapp/src/event/EventConnected.tests.js
+++ /dev/null
@@ -1,53 +0,0 @@
-import React from 'react';
-
-import Providers from 'apolloschurchapp/src/Providers';
-
-import moment from 'moment';
-import { renderWithApolloData } from '../utils/testUtils';
-import getEvent from './getEvent';
-import EventConnected from './EventConnected';
-
-const mocks = {
-  request: {
-    query: getEvent,
-    variables: { eventId: 'Event:123' },
-  },
-  result: {
-    data: {
-      node: {
-        __typename: 'Event',
-        id: 'Event:123',
-        start: moment('2019-09-26T15:10:51.200Z')
-          .utc()
-          .toJSON(),
-        end: moment('2019-09-26T17:10:51.200Z')
-          .utc()
-          .toJSON(),
-        location: 'Willow Creek, Chicago',
-        name: 'Cookout',
-        description: 'Some detailed description',
-        image: {
-          __typename: 'ImageMedia',
-          sources: [
-            {
-              uri: 'https://url.com/image.jpg',
-              __typename: 'ImageMediaSource',
-            },
-          ],
-        },
-      },
-    },
-  },
-};
-
-describe('EventConnected component', () => {
-  it('renders without errors', async () => {
-    const tree = await renderWithApolloData(
-      <Providers mocks={[mocks]}>
-        <EventConnected navigation={{ getParam: () => 'Event:123' }} />
-      </Providers>
-    );
-
-    expect(tree).toMatchSnapshot();
-  });
-});
diff --git a/apolloschurchapp/src/event/components.js b/apolloschurchapp/src/event/components.js
deleted file mode 100644
index 509e400..0000000
--- a/apolloschurchapp/src/event/components.js
+++ /dev/null
@@ -1,44 +0,0 @@
-import React from 'react';
-import { ScrollView, View } from 'react-native';
-import PropTypes from 'prop-types';
-import { styled, withTheme, H6, H4, Icon } from '@apollosproject/ui-kit';
-
-const FlexedScrollView = styled({ flex: 1 })(ScrollView);
-
-const StyledIcon = withTheme(({ theme: { colors, sizing } }) => ({
-  fill: colors.text.tertiary,
-  size: sizing.baseUnit * 1.5,
-  style: {
-    marginRight: 8,
-  },
-}))(Icon);
-
-const StyledH6 = styled(({ theme: { colors, sizing } }) => ({
-  color: colors.text.tertiary,
-  fontSize: sizing.baseUnit * 0.875,
-  marginTop: sizing.baseUnit / 4,
-}))(H6);
-
-const EventInfoContainer = styled({ marginBottom: 24, flexDirection: 'row' })(
-  View
-);
-
-const TextContainer = styled({ flexDirection: 'column' })(View);
-
-const EventInfoItem = ({ icon, title, subtitle }) => (
-  <EventInfoContainer>
-    <StyledIcon name={icon} />
-    <TextContainer>
-      <H4 bold>{title}</H4>
-      <StyledH6>{subtitle}</StyledH6>
-    </TextContainer>
-  </EventInfoContainer>
-);
-
-EventInfoItem.propTypes = {
-  icon: PropTypes.string.isRequired,
-  title: PropTypes.string,
-  subtitle: PropTypes.string,
-};
-
-export { FlexedScrollView, EventInfoItem };
diff --git a/apolloschurchapp/src/event/getEvent.js b/apolloschurchapp/src/event/getEvent.js
deleted file mode 100644
index 059cb9f..0000000
--- a/apolloschurchapp/src/event/getEvent.js
+++ /dev/null
@@ -1,29 +0,0 @@
-import gql from 'graphql-tag';
-
-export const EVENT_FRAGMENT = gql`
-  fragment eventFragment on Event {
-    id
-    name
-    location
-    start
-    end
-    description
-    image {
-      sources {
-        uri
-      }
-    }
-  }
-`;
-
-export default gql`
-  query getEvent($eventId: ID!) {
-    node(id: $eventId) {
-      __typename
-      ... on Event {
-        ...eventFragment
-      }
-    }
-  }
-  ${EVENT_FRAGMENT}
-`;
diff --git a/apolloschurchapp/src/event/index.js b/apolloschurchapp/src/event/index.js
deleted file mode 100644
index b065c4c..0000000
--- a/apolloschurchapp/src/event/index.js
+++ /dev/null
@@ -1,17 +0,0 @@
-import { createStackNavigator } from 'react-navigation';
-
-import Event from './EventConnected';
-
-const EventNavigator = createStackNavigator(
-  {
-    Event,
-  },
-  {
-    initialRouteName: 'Event',
-    headerMode: 'float',
-    headerTransitionPreset: 'fade-in-place',
-    navigationOptions: { header: null },
-  }
-);
-
-export default EventNavigator;
diff --git a/apolloschurchapp/src/index.js b/apolloschurchapp/src/index.js
index 114240e..3d63ed2 100644
--- a/apolloschurchapp/src/index.js
+++ b/apolloschurchapp/src/index.js
@@ -12,7 +12,6 @@ import Auth, { ProtectedRoute } from '@apollosproject/ui-auth';
 import Providers from './Providers';
 import NavigationService from './NavigationService';
 import ContentSingle from './content-single';
-import Event from './event';
 import Tabs from './tabs';
 import PersonalDetails from './user-settings/PersonalDetails';
 import ChangePassword from './user-settings/ChangePassword';
@@ -42,7 +41,6 @@ const AppNavigator = createStackNavigator(
     ProtectedRoute: ProtectedRouteWithSplashScreen,
     Tabs,
     ContentSingle,
-    Event,
     Auth: EnhancedAuth,
     PersonalDetails,
     ChangePassword,
diff --git a/apolloschurchapp/src/tabs/connect/RecentlyLikedTileFeed/RecentlyLikedTileFeedConnected.tests.js b/apolloschurchapp/src/tabs/connect/RecentlyLikedTileFeed/RecentlyLikedTileFeedConnected.tests.js
index 016b246..a8620c1 100644
--- a/apolloschurchapp/src/tabs/connect/RecentlyLikedTileFeed/RecentlyLikedTileFeedConnected.tests.js
+++ b/apolloschurchapp/src/tabs/connect/RecentlyLikedTileFeed/RecentlyLikedTileFeedConnected.tests.js
@@ -36,7 +36,6 @@ describe('RecentlyLikedTileFeedConnected', () => {
                     __typename: 'ContentChannel',
                   },
                   title: 'Mea Animal Aperiam Ornatus Eu',
-                  hyphenatedTitle: 'Mea Animal Aperiam Ornatus Eu',
                   summary: 'Bla bla bla',
                   theme: null,
                   isLiked: false,
@@ -58,8 +57,6 @@ describe('RecentlyLikedTileFeedConnected', () => {
                     __typename: 'ContentChannel',
                   },
                   title: 'Probo Senserit Id Mea, Ut Sed Malis Postea,',
-                  hyphenatedTitle:
-                    'Probo Senserit Id Mea, Ut Sed Malis Postea,',
                   summary: 'Bla bla bla',
                   theme: null,
                   isLiked: false,
diff --git a/apolloschurchapp/src/tabs/discover/DiscoverFeed/DiscoverFeed.tests.js b/apolloschurchapp/src/tabs/discover/DiscoverFeed/DiscoverFeed.tests.js
index 568ee71..3ce5dec 100644
--- a/apolloschurchapp/src/tabs/discover/DiscoverFeed/DiscoverFeed.tests.js
+++ b/apolloschurchapp/src/tabs/discover/DiscoverFeed/DiscoverFeed.tests.js
@@ -29,7 +29,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:00bb7b364911281c97fc50f2a0d17b11',
                       title: 'Anderson Family Cookout!',
-                      hyphenatedTitle: 'Anderson Family Cookout!',
                       isLiked: false,
                       likedCount: 0,
                       summary: 'Come join us!',
@@ -61,8 +60,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:7db601b238aa4345a818654300f83dd2',
                       title: 'Robert Madu is Coming to Christ Fellowship',
-                      hyphenatedTitle:
-                        'Robert Madu is Coming to Christ Fellowship',
                       isLiked: false,
                       likedCount: 5,
                       summary:
@@ -95,7 +92,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:296373ecb53580855cadffa0375ebe18',
                       title: 'Guys Night!',
-                      hyphenatedTitle: 'Guys Night!',
                       isLiked: false,
                       likedCount: 0,
                       summary:
@@ -138,7 +134,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:604a4a763a5fc0edfe386684ca6b3515',
                       title: 'Test Article',
-                      hyphenatedTitle: 'Test Article',
                       isLiked: false,
                       likedCount: 0,
                       summary: 'Test article...',
@@ -160,8 +155,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:c495ff18cd998ed516a798b6218907cd',
                       title: '10 ways to refresh your spirit every day',
-                      hyphenatedTitle:
-                        '10 ways to refresh your spirit every day',
                       isLiked: false,
                       likedCount: 2,
                       summary:
@@ -184,7 +177,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
                       title: 'Live for Freedom Toolkit',
-                      hyphenatedTitle: 'Live for Freedom Toolkit',
                       isLiked: false,
                       likedCount: 1,
                       summary:
@@ -217,7 +209,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'ContentSeriesContentItem:6d8dc8e4bad017f815ae0cac8bf692bb',
                       title: 'The New Weekend Experience',
-                      hyphenatedTitle: 'The New Weekend Experience',
                       isLiked: false,
                       likedCount: 1,
                       summary:
@@ -253,7 +244,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'ContentSeriesContentItem:08449058d438ebeaffe9adb7a8e633cc',
                       title: 'Gathering',
-                      hyphenatedTitle: 'Gathering',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -288,7 +278,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'ContentSeriesContentItem:b74c2d295a0cf1340232ffa8be856c00',
                       title: 'Gauntlet 2018',
-                      hyphenatedTitle: 'Gauntlet 2018',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -333,7 +322,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'ContentSeriesContentItem:b085ec27d53006e9e6b95f3ac57da581',
                       title: 'Jeremiah: A 33-Day Devotional',
-                      hyphenatedTitle: 'Jeremiah: A 33-Day Devotional',
                       isLiked: true,
                       likedCount: 6,
                       summary: 'Have you ever had to deliver difficult news?',
@@ -365,8 +353,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'ContentSeriesContentItem:71bfdac0b98ea1b72a63ff4ea64e3c5a',
                       title: 'How To Be A Better Leader: A 7-Day Devotional',
-                      hyphenatedTitle:
-                        'How To Be A Better Leader: A 7-Day Devotional',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -398,7 +384,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'ContentSeriesContentItem:44fd3b82373ea8f197bd1e1e77c4bed0',
                       title: '2 Thessalonians: A 5-Day Devotional',
-                      hyphenatedTitle: '2 Thessalonians: A 5-Day Devotional',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -441,8 +426,6 @@ describe('The DiscoverFeed component', () => {
                         'UniversalContentItem:cf8fa657f23af88a2d63f7e4a2237db8',
                       title:
                         'The surprising journey that changed me and my dad',
-                      hyphenatedTitle:
-                        'The surprising journey that changed me and my dad',
                       isLiked: false,
                       likedCount: 6,
                       summary:
@@ -475,8 +458,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:2e17f5a66407d114ab9f2392b03ebccd',
                       title: 'When I lost hope, my church family had my back',
-                      hyphenatedTitle:
-                        'When I lost hope, my church family had my back',
                       isLiked: true,
                       likedCount: 5,
                       summary:
@@ -509,7 +490,6 @@ describe('The DiscoverFeed component', () => {
                       id:
                         'UniversalContentItem:ae8ec75906ba7437c49ad2534b5024db',
                       title: 'A Place to Worship Free of Fear',
-                      hyphenatedTitle: 'A Place to Worship Free of Fear',
                       isLiked: false,
                       likedCount: 1,
                       summary:
diff --git a/apolloschurchapp/src/tabs/discover/SearchFeed/SearchFeed.tests.js b/apolloschurchapp/src/tabs/discover/SearchFeed/SearchFeed.tests.js
index aed2839..6dbc4da 100644
--- a/apolloschurchapp/src/tabs/discover/SearchFeed/SearchFeed.tests.js
+++ b/apolloschurchapp/src/tabs/discover/SearchFeed/SearchFeed.tests.js
@@ -40,7 +40,6 @@ describe('The SearchFeed component', () => {
                 node: {
                   id: 'DevotionalContentItem:561dfb7dbd8a5c093fd8385c7edaadbc',
                   title: 'How to lead people to Jesus',
-                  hyphenatedTitle: '-',
                   isLiked: false,
                   likedCount: 0,
                   summary:
@@ -77,7 +76,6 @@ describe('The SearchFeed component', () => {
                 node: {
                   id: 'UniversalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
                   title: 'Live for Freedom Toolkit',
-                  hyphenatedTitle: '-',
                   isLiked: false,
                   likedCount: 1,
                   summary:
@@ -114,7 +112,6 @@ describe('The SearchFeed component', () => {
                 node: {
                   id: 'UniversalContentItem:895738f70482712adb3ab45a08c30456',
                   title: 'Should your 20-something live at home?',
-                  hyphenatedTitle: '-',
                   isLiked: false,
                   likedCount: 0,
                   summary:
@@ -160,7 +157,6 @@ describe('The SearchFeed component', () => {
                 node: {
                   id: 'UniversalContentItem:296373ecb53580855cadffa0375ebe18',
                   title: 'Guys Night!',
-                  hyphenatedTitle: '-',
                   isLiked: false,
                   likedCount: 0,
                   summary: 'A time to connect with other guys who live nearby.',
@@ -196,7 +192,6 @@ describe('The SearchFeed component', () => {
                 node: {
                   id: 'UniversalContentItem:c495ff18cd998ed516a798b6218907cd',
                   title: '10 ways to refresh your spirit every day',
-                  hyphenatedTitle: '-',
                   isLiked: false,
                   likedCount: 2,
                   summary:
@@ -233,7 +228,6 @@ describe('The SearchFeed component', () => {
                 node: {
                   id: 'DevotionalContentItem:4dd74ceca9e09b5276e93cbf53821bd6',
                   title: 'Why leadership isn’t all about the title',
-                  hyphenatedTitle: '-',
                   isLiked: false,
                   likedCount: 0,
                   summary:
diff --git a/apolloschurchapp/src/tabs/discover/SearchFeed/index.js b/apolloschurchapp/src/tabs/discover/SearchFeed/index.js
index 24d2a6e..63a7e16 100644
--- a/apolloschurchapp/src/tabs/discover/SearchFeed/index.js
+++ b/apolloschurchapp/src/tabs/discover/SearchFeed/index.js
@@ -34,9 +34,7 @@ const SearchFeed = withNavigation(({ navigation, searchText }) => (
     {({ loading, error, data, refetch }) => (
       <StyledFeedView
         ListItemComponent={ContentCardConnected}
-        content={get(data, 'search.edges', [])
-          .map((edge) => edge.node)
-          .filter((node) => !!node)}
+        content={get(data, 'search.edges', []).map((edge) => edge.node)}
         ListEmptyComponent={() => <NoResults searchText={searchText} />}
         hasContent={get(data, 'search.edges', []).length}
         isLoading={loading}
diff --git a/apolloschurchapp/src/tabs/discover/discover.tests.js b/apolloschurchapp/src/tabs/discover/discover.tests.js
index abd60dd..93fd091 100644
--- a/apolloschurchapp/src/tabs/discover/discover.tests.js
+++ b/apolloschurchapp/src/tabs/discover/discover.tests.js
@@ -29,7 +29,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:00bb7b364911281c97fc50f2a0d17b11',
                       title: 'Anderson Family Cookout!',
-                      hyphenatedTitle: 'Anderson Family Cookout!',
                       isLiked: false,
                       likedCount: 0,
                       summary: 'Come join us!',
@@ -61,8 +60,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:7db601b238aa4345a818654300f83dd2',
                       title: 'Robert Madu is Coming to Christ Fellowship',
-                      hyphenatedTitle:
-                        'Robert Madu is Coming to Christ Fellowship',
                       isLiked: false,
                       likedCount: 5,
                       summary:
@@ -95,7 +92,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:296373ecb53580855cadffa0375ebe18',
                       title: 'Guys Night!',
-                      hyphenatedTitle: 'Guys Night!',
                       isLiked: false,
                       likedCount: 0,
                       summary:
@@ -138,7 +134,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:604a4a763a5fc0edfe386684ca6b3515',
                       title: 'Test Article',
-                      hyphenatedTitle: 'Test Article',
                       isLiked: false,
                       likedCount: 0,
                       summary: 'Test article...',
@@ -160,8 +155,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:c495ff18cd998ed516a798b6218907cd',
                       title: '10 ways to refresh your spirit every day',
-                      hyphenatedTitle:
-                        '10 ways to refresh your spirit every day',
                       isLiked: false,
                       likedCount: 2,
                       summary:
@@ -184,7 +177,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
                       title: 'Live for Freedom Toolkit',
-                      hyphenatedTitle: 'Live for Freedom Toolkit',
                       isLiked: false,
                       likedCount: 1,
                       summary:
@@ -217,7 +209,6 @@ describe('The Discover tab component', () => {
                       id:
                         'ContentSeriesContentItem:6d8dc8e4bad017f815ae0cac8bf692bb',
                       title: 'The New Weekend Experience',
-                      hyphenatedTitle: 'The New Weekend Experience',
                       isLiked: false,
                       likedCount: 1,
                       summary:
@@ -253,7 +244,6 @@ describe('The Discover tab component', () => {
                       id:
                         'ContentSeriesContentItem:08449058d438ebeaffe9adb7a8e633cc',
                       title: 'Gathering',
-                      hyphenatedTitle: 'Gathering',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -288,7 +278,6 @@ describe('The Discover tab component', () => {
                       id:
                         'ContentSeriesContentItem:b74c2d295a0cf1340232ffa8be856c00',
                       title: 'Gauntlet 2018',
-                      hyphenatedTitle: 'Gauntlet 2018',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -333,7 +322,6 @@ describe('The Discover tab component', () => {
                       id:
                         'ContentSeriesContentItem:b085ec27d53006e9e6b95f3ac57da581',
                       title: 'Jeremiah: A 33-Day Devotional',
-                      hyphenatedTitle: 'Jeremiah: A 33-Day Devotional',
                       isLiked: true,
                       likedCount: 6,
                       summary: 'Have you ever had to deliver difficult news?',
@@ -365,8 +353,6 @@ describe('The Discover tab component', () => {
                       id:
                         'ContentSeriesContentItem:71bfdac0b98ea1b72a63ff4ea64e3c5a',
                       title: 'How To Be A Better Leader: A 7-Day Devotional',
-                      hyphenatedTitle:
-                        'How To Be A Better Leader: A 7-Day Devotional',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -398,7 +384,6 @@ describe('The Discover tab component', () => {
                       id:
                         'ContentSeriesContentItem:44fd3b82373ea8f197bd1e1e77c4bed0',
                       title: '2 Thessalonians: A 5-Day Devotional',
-                      hyphenatedTitle: '2 Thessalonians: A 5-Day Devotional',
                       isLiked: false,
                       likedCount: 0,
                       summary: '',
@@ -441,8 +426,6 @@ describe('The Discover tab component', () => {
                         'UniversalContentItem:cf8fa657f23af88a2d63f7e4a2237db8',
                       title:
                         'The surprising journey that changed me and my dad',
-                      hyphenatedTitle:
-                        'The surprising journey that changed me and my dad',
                       isLiked: false,
                       likedCount: 6,
                       summary:
@@ -475,8 +458,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:2e17f5a66407d114ab9f2392b03ebccd',
                       title: 'When I lost hope, my church family had my back',
-                      hyphenatedTitle:
-                        'When I lost hope, my church family had my back',
                       isLiked: true,
                       likedCount: 5,
                       summary:
@@ -509,7 +490,6 @@ describe('The Discover tab component', () => {
                       id:
                         'UniversalContentItem:ae8ec75906ba7437c49ad2534b5024db',
                       title: 'A Place to Worship Free of Fear',
-                      hyphenatedTitle: 'A Place to Worship Free of Fear',
                       isLiked: false,
                       likedCount: 1,
                       summary:
diff --git a/apolloschurchapp/src/tabs/home/Features/index.js b/apolloschurchapp/src/tabs/home/Features/index.js
index 80aebcb..87878f3 100644
--- a/apolloschurchapp/src/tabs/home/Features/index.js
+++ b/apolloschurchapp/src/tabs/home/Features/index.js
@@ -109,12 +109,6 @@ const Features = memo(({ navigation }) => (
                       transitionKey: 2,
                     });
                   }
-                  if (action === 'READ_EVENT') {
-                    navigation.navigate('Event', {
-                      eventId: relatedNode.id,
-                      transitionKey: 2,
-                    });
-                  }
                 }}
                 onPressCardActionButton={() =>
                   navigation.navigate('ContentFeed', {
diff --git a/apolloschurchapp/src/testing-control-panel/index.js b/apolloschurchapp/src/testing-control-panel/index.js
index 79030be..6a6c16a 100644
--- a/apolloschurchapp/src/testing-control-panel/index.js
+++ b/apolloschurchapp/src/testing-control-panel/index.js
@@ -1,7 +1,6 @@
 import React, { PureComponent } from 'react';
 import { TableView, Divider } from '@apollosproject/ui-kit';
 import { UserWebBrowserConsumer } from 'apolloschurchapp/src/user-web-browser';
-import { WebBrowserConsumer } from 'apolloschurchapp/src/ui/WebBrowser';
 import ChangeLivestream from './ChangeLivestream';
 import TouchableCell from './TouchableCell';
 
@@ -29,21 +28,6 @@ export default class TestingControlPanel extends PureComponent {
             />
           )}
         </UserWebBrowserConsumer>
-        <WebBrowserConsumer>
-          {(openUrl) => (
-            <TouchableCell
-              handlePress={() =>
-                openUrl(
-                  'https://apollosrock.newspring.cc',
-                  {},
-                  { useRockToken: true }
-                )
-              }
-              iconName="share"
-              cellText={`Open InAppBrowser With Rock Token`}
-            />
-          )}
-        </WebBrowserConsumer>
         <TouchableCell
           handlePress={() => this.props.navigation.navigate('Onboarding')}
           iconName="Avatar"
diff --git a/apolloschurchapp/src/ui/ContentCardConnected/query.js b/apolloschurchapp/src/ui/ContentCardConnected/query.js
index 8b7d514..1ff178a 100644
--- a/apolloschurchapp/src/ui/ContentCardConnected/query.js
+++ b/apolloschurchapp/src/ui/ContentCardConnected/query.js
@@ -31,7 +31,6 @@ export const BASE_CARD_FRAGMENT = gql`
     ...coverImageFragment
     ...themeFragment
     title
-    hyphenatedTitle: title(hyphenated: true)
     summary
     isLiked
     ... on MediaContentItem {
diff --git a/apolloschurchapp/src/ui/HorizontalContentCardConnected/horizontalContentCardComponentMapper.js b/apolloschurchapp/src/ui/HorizontalContentCardConnected/horizontalContentCardComponentMapper.js
index 433b143..c658e47 100644
--- a/apolloschurchapp/src/ui/HorizontalContentCardConnected/horizontalContentCardComponentMapper.js
+++ b/apolloschurchapp/src/ui/HorizontalContentCardConnected/horizontalContentCardComponentMapper.js
@@ -1,32 +1,22 @@
 import React from 'react';
 import { get } from 'lodash';
-import PropTypes from 'prop-types';
 
 import {
   HorizontalDefaultCard,
   HorizontalHighlightCard,
 } from '@apollosproject/ui-kit';
 
-const horizontalContentCardComponentMapper = ({
-  title,
-  hyphenatedTitle,
-  ...props
-}) => {
+const horizontalContentCardComponentMapper = (props) => {
   // map typename to the the card we want to render.
   switch (get(props, '__typename')) {
     case 'MediaContentItem':
     case 'WeekendContentItem':
     case 'ContentSeriesContentItem':
     case 'DevotionalContentItem':
-      return <HorizontalHighlightCard title={hyphenatedTitle} {...props} />;
+      return <HorizontalHighlightCard {...props} />;
     default:
-      return <HorizontalDefaultCard title={title} {...props} />;
+      return <HorizontalDefaultCard {...props} />;
   }
 };
 
-horizontalContentCardComponentMapper.propTypes = {
-  hyphenatedTitle: PropTypes.string,
-  title: PropTypes.string,
-};
-
 export default horizontalContentCardComponentMapper;
diff --git a/apolloschurchapp/src/ui/WebBrowser/Browser.js b/apolloschurchapp/src/ui/WebBrowser/Browser.js
index 42a7766..c6ddcb8 100644
--- a/apolloschurchapp/src/ui/WebBrowser/Browser.js
+++ b/apolloschurchapp/src/ui/WebBrowser/Browser.js
@@ -3,51 +3,39 @@ import InAppBrowser from 'react-native-inappbrowser-reborn';
 import gql from 'graphql-tag';
 import { client } from '../../client';
 
-export const GET_ROCK_AUTH_DETAILS = gql`
-  query RockAuthDetails {
+export const GET_USER_COOKIE = gql`
+  query CurrentUserCookie {
     currentUser {
       id
-      rock {
-        authCookie
-        authToken
-      }
+      rockToken
     }
   }
 `;
 
-export const getRockAuthDetails = async () => {
-  const { data: { currentUser: { rock } = {} } = {} } = await client.query({
-    query: GET_ROCK_AUTH_DETAILS,
-    fetchPolicy: 'network-only',
-  });
-  return rock;
+export const getCookie = async () => {
+  const { data: { currentUser: { rockToken } = {} } = {} } = await client.query(
+    {
+      query: GET_USER_COOKIE,
+    }
+  );
+  return rockToken;
 };
 
 const Browser = {
-  open: async (
-    baseURL,
-    options,
-    auth = { useRockCookie: false, useRockToken: false }
-  ) => {
-    const url = new URL(baseURL);
-    const { authCookie, authToken } = await getRockAuthDetails();
-    let headers = {};
-    if (auth.useRockCookie && authCookie) {
+  open: async (url, options) => {
+    const cookie = await getCookie();
+    if (cookie)
       console.warn(
         "iOS doesn't support headers, you may want to use src/user-web-view"
       );
-      headers = { Cookie: authCookie };
-    }
-    if (auth.useRockToken && authToken) {
-      url.searchParams.append('rckipid', authToken);
-    }
+    const headers = { Cookie: cookie };
     try {
       if (await InAppBrowser.isAvailable()) {
-        InAppBrowser.open(url.toString(), {
+        InAppBrowser.open(url, {
           headers,
           ...options,
         });
-      } else Linking.openURL(url.toString());
+      } else Linking.openURL(url);
     } catch (e) {
       console.error(e);
     }
diff --git a/apolloschurchapp/src/user-settings/ChangePassword.js b/apolloschurchapp/src/user-settings/ChangePassword.js
index b8683fc..6045e31 100644
--- a/apolloschurchapp/src/user-settings/ChangePassword.js
+++ b/apolloschurchapp/src/user-settings/ChangePassword.js
@@ -1,13 +1,7 @@
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
-import {
-  Platform,
-  KeyboardAvoidingView,
-  StyleSheet,
-  StatusBar,
-} from 'react-native';
+import { SafeAreaView } from 'react-native';
 import { Mutation } from 'react-apollo';
-import { SafeAreaView, Header } from 'react-navigation';
 import { Formik } from 'formik';
 import * as Yup from 'yup';
 
@@ -15,6 +9,7 @@ import {
   Button,
   ButtonLink,
   TextInput,
+  BackgroundView,
   PaddedView,
   FlexedView,
   styled,
@@ -48,15 +43,8 @@ class ChangePassword extends PureComponent {
   };
 
   renderForm = (props) => (
-    <KeyboardAvoidingView
-      behavior={'padding'}
-      style={StyleSheet.absoluteFill}
-      keyboardVerticalOffset={
-        Header.HEIGHT +
-        (Platform.OS === 'android' ? StatusBar.currentHeight : 0)
-      }
-    >
-      <FlexedView>
+    <FlexedView>
+      <BackgroundView>
         <PaddedView>
           <TextInput
             label="New Password"
@@ -77,18 +65,18 @@ class ChangePassword extends PureComponent {
             }
           />
         </PaddedView>
-        <Footer>
-          <PaddedView>
-            <Button
-              disabled={props.isSubmitting}
-              onPress={props.handleSubmit}
-              title="Save"
-              loading={props.isSubmitting}
-            />
-          </PaddedView>
-        </Footer>
-      </FlexedView>
-    </KeyboardAvoidingView>
+      </BackgroundView>
+      <Footer>
+        <PaddedView>
+          <Button
+            disabled={props.isSubmitting}
+            onPress={props.handleSubmit}
+            title="Save"
+            loading={props.isSubmitting}
+          />
+        </PaddedView>
+      </Footer>
+    </FlexedView>
   );
 
   render() {
diff --git a/apolloschurchapp/src/user-settings/Locations/Locations.tests.js b/apolloschurchapp/src/user-settings/Locations/Locations.tests.js
index 25ebf12..9e131ae 100644
--- a/apolloschurchapp/src/user-settings/Locations/Locations.tests.js
+++ b/apolloschurchapp/src/user-settings/Locations/Locations.tests.js
@@ -110,15 +110,6 @@ describe('Display Native Map with Locations', () => {
       result: {
         data: {
           campuses,
-          currentUser: {
-            id: 'AuthenticatedUser:123',
-            __typename: 'AuthenticatedUser',
-            profile: {
-              id: 'Person:123',
-              __typename: 'Person',
-              campus: campuses[3],
-            },
-          },
         },
       },
     };
diff --git a/apolloschurchapp/src/user-settings/Locations/MapView.js b/apolloschurchapp/src/user-settings/Locations/MapView.js
index 7f9b2bc..9478cf7 100644
--- a/apolloschurchapp/src/user-settings/Locations/MapView.js
+++ b/apolloschurchapp/src/user-settings/Locations/MapView.js
@@ -7,7 +7,6 @@ import { debounce } from 'lodash';
 
 import {
   Button,
-  Touchable,
   PaddedView,
   FlexedView,
   styled,
@@ -48,10 +47,6 @@ class MapView extends Component {
         longitude: PropTypes.number.isRequired,
       })
     ),
-    currentCampus: PropTypes.shape({
-      latitude: PropTypes.number.isRequired,
-      longitude: PropTypes.number.isRequired,
-    }),
     onLocationSelect: PropTypes.func.isRequired,
     initialRegion: PropTypes.shape({
       latitude: PropTypes.number.isRequired,
@@ -75,8 +70,6 @@ class MapView extends Component {
 
   animation = new Animated.Value(0);
 
-  scrollView = null;
-
   componentDidMount() {
     this.animation.addListener(debounce(this.updateCoordinates));
   }
@@ -91,33 +84,13 @@ class MapView extends Component {
     const cardIndex = Math.floor(
       this.previousScrollPosition / CARD_WIDTH + 0.3
     ); // animate 30% away from landing on the next item;
-    return this.sortedCampuses[cardIndex];
-  }
-
-  get sortedCampuses() {
-    const { currentCampus = null, campuses = [] } = this.props;
-    if (!this.props.currentCampus) {
-      return campuses;
-    }
-    return [
-      currentCampus,
-      ...campuses.filter(({ id }) => id !== currentCampus.id),
-    ];
+    return this.props.campuses[cardIndex];
   }
 
-  scrollToIndex = (index) => {
-    this.scrollView.getNode().scrollTo({
-      x: index * (CARD_WIDTH + 8),
-      y: 0,
-      animated: true,
-    });
-    this.updateCoordinates({ value: index * (CARD_WIDTH + 8) });
-  };
-
   updateCoordinates = ({ value }) => {
     this.previousScrollPosition = value;
 
-    const { userLocation } = this.props;
+    const { userLocation, campuses } = this.props;
     // campus card height + some padding
 
     const bottomPadding = 100 + this.props.theme.sizing.baseUnit * 12;
@@ -135,7 +108,7 @@ class MapView extends Component {
 
     const visibleCampuses = [
       userLocation,
-      ...(this.currentCampus ? [this.currentCampus] : this.sortedCampuses),
+      ...(this.currentCampus ? [this.currentCampus] : campuses),
     ];
 
     this.map.fitToCoordinates(visibleCampuses, {
@@ -144,8 +117,8 @@ class MapView extends Component {
   };
 
   render() {
-    const { onLocationSelect } = this.props;
-    const interpolations = this.sortedCampuses.map((marker, index) => {
+    const { campuses = [], onLocationSelect } = this.props;
+    const interpolations = campuses.map((marker, index) => {
       const inputRange = [
         (index - 1) * CARD_WIDTH,
         index * CARD_WIDTH,
@@ -168,13 +141,12 @@ class MapView extends Component {
             this.map = map;
           }}
         >
-          {this.sortedCampuses.map((campus, index) => {
+          {campuses.map((campus, index) => {
             const campusOpacity = {
               opacity: interpolations[index].opacity,
             };
             return (
               <Marker
-                onPress={() => this.scrollToIndex(index)}
                 key={campus.id}
                 opacityStyle={campusOpacity}
                 latitude={campus.latitude}
@@ -193,7 +165,6 @@ class MapView extends Component {
             contentContainerStyle={{
               paddingHorizontal: this.props.theme.sizing.baseUnit * 0.75,
             }}
-            ref={(ref) => (this.scrollView = ref)} // eslint-disable-line
             scrollEventThrottle={16} // roughtly 1000ms/60fps = 16ms
             onScroll={Animated.event(
               [
@@ -208,18 +179,14 @@ class MapView extends Component {
               { useNativeDriver: true }
             )}
           >
-            {this.sortedCampuses.map((campus) => (
-              <Touchable
+            {campuses.map((campus) => (
+              <StyledCampusCard
                 key={campus.id}
-                onPress={() => onLocationSelect(campus)}
-              >
-                <StyledCampusCard
-                  distance={campus.distanceFromLocation}
-                  title={campus.name}
-                  description={getCampusAddress(campus)}
-                  images={[campus.image]}
-                />
-              </Touchable>
+                distance={campus.distanceFromLocation}
+                title={campus.name}
+                description={getCampusAddress(campus)}
+                images={[campus.image]}
+              />
             ))}
           </Animated.ScrollView>
           <PaddedView>
@@ -228,7 +195,7 @@ class MapView extends Component {
               pill={false}
               type="secondary"
               onPress={() =>
-                onLocationSelect(this.currentCampus || this.sortedCampuses[0])
+                onLocationSelect(this.currentCampus || campuses[0])
               }
             />
           </PaddedView>
diff --git a/apolloschurchapp/src/user-settings/Locations/Marker.js b/apolloschurchapp/src/user-settings/Locations/Marker.js
index b91f89e..0c98fa9 100644
--- a/apolloschurchapp/src/user-settings/Locations/Marker.js
+++ b/apolloschurchapp/src/user-settings/Locations/Marker.js
@@ -25,8 +25,8 @@ const MarkerRingView = styled(({ theme }) => ({
   justifyContent: 'center',
 }))(View);
 
-const StyledMarker = memo(({ latitude, longitude, opacityStyle, onPress }) => (
-  <Marker onPress={onPress} coordinate={{ latitude, longitude }}>
+const StyledMarker = memo(({ latitude, longitude, opacityStyle }) => (
+  <Marker coordinate={{ latitude, longitude }}>
     <Animated.View style={opacityStyle}>
       <MarkerRingView>
         <MarkerView />
@@ -40,7 +40,6 @@ StyledMarker.displayName = 'StyledMarker';
 StyledMarker.propTypes = {
   latitude: PropTypes.number.isRequired,
   longitude: PropTypes.number.isRequired,
-  onPress: PropTypes.func.isRequired,
   opacityStyle: PropTypes.shape({}),
 };
 
diff --git a/apolloschurchapp/src/user-settings/Locations/getCampusLocations.js b/apolloschurchapp/src/user-settings/Locations/getCampusLocations.js
index 6dc73e9..1c98bb8 100644
--- a/apolloschurchapp/src/user-settings/Locations/getCampusLocations.js
+++ b/apolloschurchapp/src/user-settings/Locations/getCampusLocations.js
@@ -6,15 +6,6 @@ export default gql`
     campuses(location: { latitude: $latitude, longitude: $longitude }) {
       ...CampusParts
     }
-    currentUser {
-      id
-      profile {
-        id
-        campus {
-          ...CampusParts
-        }
-      }
-    }
   }
   ${CampusFragment}
 `;
diff --git a/apolloschurchapp/src/user-settings/Locations/index.js b/apolloschurchapp/src/user-settings/Locations/index.js
index d54cc06..985651a 100644
--- a/apolloschurchapp/src/user-settings/Locations/index.js
+++ b/apolloschurchapp/src/user-settings/Locations/index.js
@@ -4,7 +4,7 @@ import { Query, Mutation } from 'react-apollo';
 import { Dimensions } from 'react-native';
 import Geolocation from 'react-native-geolocation-service';
 import { PaddedView, ButtonLink } from '@apollosproject/ui-kit';
-import { get } from 'lodash';
+import requestLocation from './requestLocation';
 
 import GET_CAMPUSES from './getCampusLocations';
 import CHANGE_CAMPUS from './campusChange';
@@ -55,6 +55,7 @@ class Location extends PureComponent {
   };
 
   async componentDidMount() {
+    await requestLocation();
     Geolocation.getCurrentPosition(
       (position) => {
         this.setState({
@@ -79,7 +80,7 @@ class Location extends PureComponent {
         }}
         fetchPolicy="cache-and-network"
       >
-        {({ loading, error, data: { campuses = [], currentUser } = {} }) => (
+        {({ loading, error, data: { campuses = [] } = {} }) => (
           <Mutation mutation={CHANGE_CAMPUS}>
             {(handlePress) => (
               <MapView
@@ -89,7 +90,6 @@ class Location extends PureComponent {
                 campuses={campuses}
                 initialRegion={this.props.initialRegion}
                 userLocation={this.state.userLocation}
-                currentCampus={get(currentUser, 'profile.campus')}
                 onLocationSelect={async ({ id }) => {
                   await handlePress({
                     variables: {
diff --git a/apolloschurchapp/src/user-settings/Locations/requestLocation.android.js b/apolloschurchapp/src/user-settings/Locations/requestLocation.android.js
new file mode 100644
index 0000000..02c84c6
--- /dev/null
+++ b/apolloschurchapp/src/user-settings/Locations/requestLocation.android.js
@@ -0,0 +1,6 @@
+import { PermissionsAndroid } from 'react-native';
+
+export default async () =>
+  PermissionsAndroid.request(
+    PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION
+  );
diff --git a/apolloschurchapp/src/user-settings/Locations/requestLocation.ios.js b/apolloschurchapp/src/user-settings/Locations/requestLocation.ios.js
new file mode 100644
index 0000000..88ce139
--- /dev/null
+++ b/apolloschurchapp/src/user-settings/Locations/requestLocation.ios.js
@@ -0,0 +1,3 @@
+// not neccessary in ios
+// this is handled by Geolocation.getCurrentPosition()
+export default () => null;
diff --git a/apolloschurchapp/src/user-settings/index.js b/apolloschurchapp/src/user-settings/index.js
index 3dfc18c..491b707 100644
--- a/apolloschurchapp/src/user-settings/index.js
+++ b/apolloschurchapp/src/user-settings/index.js
@@ -3,7 +3,6 @@ import { ScrollView } from 'react-native';
 import { StackActions, NavigationActions } from 'react-navigation';
 import PropTypes from 'prop-types';
 import { Query, Mutation } from 'react-apollo';
-import { getVersion, getBuildNumber } from 'react-native-device-info';
 
 import {
   BackgroundView,
@@ -152,13 +151,6 @@ class UserSettings extends PureComponent {
                           )}
                         </Mutation>
                       </TableView>
-                      <TableView>
-                        <Cell>
-                          <CellText>
-                            {`App Version: ${getVersion()}.${getBuildNumber()}`}
-                          </CellText>
-                        </Cell>
-                      </TableView>
                     </>
                   )}
                 </WebBrowserConsumer>
diff --git a/apolloschurchapp/src/user-settings/uploadPhoto.js b/apolloschurchapp/src/user-settings/uploadPhoto.js
index c1cc667..7eb5815 100644
--- a/apolloschurchapp/src/user-settings/uploadPhoto.js
+++ b/apolloschurchapp/src/user-settings/uploadPhoto.js
@@ -9,8 +9,6 @@ const options = {
   storageOptions: {
     skipBackup: true,
     path: 'images',
-    waitUntilSaved: true,
-    cameraRoll: true,
   },
 };
 
