diff --git a/apolloschurchapp/android/app/build.gradle b/apolloschurchapp/android/app/build.gradle
index 96db8c6..7c9c702 100644
--- a/apolloschurchapp/android/app/build.gradle
+++ b/apolloschurchapp/android/app/build.gradle
@@ -147,7 +147,7 @@ android {
         minSdkVersion rootProject.ext.minSdkVersion
         targetSdkVersion rootProject.ext.targetSdkVersion
         versionCode 24
-        versionName "1.3.1"
+        versionName "1.4.0"
         ndk {
             abiFilters "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
         }
diff --git a/apolloschurchapp/apollos.json b/apolloschurchapp/apollos.json
index a2c2d8a..fd2c7e7 100644
--- a/apolloschurchapp/apollos.json
+++ b/apolloschurchapp/apollos.json
@@ -1,4 +1,4 @@
 {
-    "version": "1.3.1",
+    "version": "1.4.0-beta.0",
     "environment": "client"
 }
\ No newline at end of file
diff --git a/apolloschurchapp/ios/Podfile.lock b/apolloschurchapp/ios/Podfile.lock
index a179625..4c2c03c 100644
--- a/apolloschurchapp/ios/Podfile.lock
+++ b/apolloschurchapp/ios/Podfile.lock
@@ -14,9 +14,9 @@ PODS:
     - glog
   - glog (0.3.5)
   - OneSignal (2.10.1)
-  - Permission-Notifications (2.0.9):
+  - Permission-Notifications (2.0.10):
     - RNPermissions
-  - Permission-PhotoLibrary (2.0.9):
+  - Permission-PhotoLibrary (2.0.10):
     - RNPermissions
   - React (0.60.5):
     - React-Core (= 0.60.5)
@@ -116,17 +116,17 @@ PODS:
     - React-Core
   - RNAirplay (1.0.0):
     - React
-  - RNBootSplash (2.0.5):
+  - RNBootSplash (2.1.0):
     - React
-  - RNCAsyncStorage (1.8.1):
+  - RNCAsyncStorage (1.9.0):
     - React
   - RNDeviceInfo (4.0.1):
     - React
   - RNGestureHandler (1.6.1):
     - React
-  - RNInAppBrowser (3.3.4):
+  - RNInAppBrowser (3.4.0):
     - React
-  - RNPermissions (2.0.9):
+  - RNPermissions (2.0.10):
     - React
   - RNScreens (1.0.0-alpha.23):
     - React
@@ -285,8 +285,8 @@ SPEC CHECKSUMS:
   Folly: 30e7936e1c45c08d884aa59369ed951a8e68cf51
   glog: 1f3da668190260b06b429bb211bfbee5cd790c28
   OneSignal: 8fb749fa07f88006403c9b659a570fb7fb74519a
-  Permission-Notifications: 1e03f5c43612ac4f9cdcaf6504695fc4e15f33d9
-  Permission-PhotoLibrary: 42d41b53adefcfa94f42fb628f143d97136120bd
+  Permission-Notifications: c9eb3a2da68dc0f37a8dc67d37bd28d150c5041d
+  Permission-PhotoLibrary: e59a7a687e002e8c8c46946c26520934ba85a688
   React: 53c53c4d99097af47cf60594b8706b4e3321e722
   React-Core: ba421f6b4f4cbe2fb17c0b6fc675f87622e78a64
   React-cxxreact: 8384287780c4999351ad9b6e7a149d9ed10a2395
@@ -317,12 +317,12 @@ SPEC CHECKSUMS:
   React-RCTWebSocket: cd932a16b7214898b6b7f788c8bddb3637246ac4
   rn-fetch-blob: 651b8d076b43d0d7aa294a3d9ec16c00aab8bef9
   RNAirplay: 75777dd55dafbd3687703c3eb9c290ea6cc43248
-  RNBootSplash: e5e37b64a03a9655e2db8c2e12a7a30853cf3352
-  RNCAsyncStorage: e0dd7c8a36543b4ef84969acd9f8aceba3a92dc2
+  RNBootSplash: fbf76a4bf1c03e9acd1dd08ce064847f26d6d8bf
+  RNCAsyncStorage: 453cd7c335ec9ba3b877e27d02238956b76f3268
   RNDeviceInfo: 687c1b2ab6d86ff1ca1208783320cd144138c7f2
   RNGestureHandler: 8f09cd560f8d533eb36da5a6c5a843af9f056b38
-  RNInAppBrowser: 9936b2e23ab7ed9b6ac938eb1c4a656b95292e11
-  RNPermissions: 2f74237e97b08beda01e914301e12524ddddf5b8
+  RNInAppBrowser: c1a0dad2734458d79bcb6c8f4297ba8658a1da9f
+  RNPermissions: 1008d3511fee0e25739cf81c4af0d1b2248f1053
   RNScreens: f28b48b8345f2f5f39ed6195518291515032a788
   RNSVG: 8ba35cbeb385a52fd960fd28db9d7d18b4c2974f
   yoga: 312528f5bbbba37b4dcea5ef00e8b4033fdd9411
diff --git a/apolloschurchapp/ios/apolloschurchapp/Info.plist b/apolloschurchapp/ios/apolloschurchapp/Info.plist
index 90c7f6c..faa3ab0 100644
--- a/apolloschurchapp/ios/apolloschurchapp/Info.plist
+++ b/apolloschurchapp/ios/apolloschurchapp/Info.plist
@@ -17,7 +17,7 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>1.3.1</string>
+	<string>1.4.0</string>
 	<key>CFBundleSignature</key>
 	<string>????</string>
 	<key>CFBundleURLTypes</key>
diff --git a/apolloschurchapp/package.json b/apolloschurchapp/package.json
index 988560c..91896ab 100644
--- a/apolloschurchapp/package.json
+++ b/apolloschurchapp/package.json
@@ -48,21 +48,21 @@
     ]
   },
   "dependencies": {
-    "@apollosproject/config": "^1.3.1",
+    "@apollosproject/config": "1.4.0-beta.0",
     "@apollosproject/react-native-airplay-btn": "0.2.0",
-    "@apollosproject/ui-analytics": "^1.3.1",
-    "@apollosproject/ui-auth": "^1.3.1",
-    "@apollosproject/ui-connected": "^1.3.1",
-    "@apollosproject/ui-fragments": "^1.3.1",
-    "@apollosproject/ui-htmlview": "^1.3.1",
-    "@apollosproject/ui-kit": "^1.3.1",
-    "@apollosproject/ui-mapview": "^1.3.1",
-    "@apollosproject/ui-media-player": "^1.3.1",
-    "@apollosproject/ui-notifications": "^1.3.1",
-    "@apollosproject/ui-onboarding": "^1.3.1",
-    "@apollosproject/ui-passes": "^1.3.1",
-    "@apollosproject/ui-scripture": "^1.3.1",
-    "@apollosproject/ui-storybook": "^1.3.1",
+    "@apollosproject/ui-analytics": "1.4.0-beta.0",
+    "@apollosproject/ui-auth": "1.4.0-beta.0",
+    "@apollosproject/ui-connected": "1.4.0-beta.0",
+    "@apollosproject/ui-fragments": "1.4.0-beta.0",
+    "@apollosproject/ui-htmlview": "1.4.0-beta.0",
+    "@apollosproject/ui-kit": "1.4.0-beta.0",
+    "@apollosproject/ui-mapview": "1.4.0-beta.0",
+    "@apollosproject/ui-media-player": "1.4.0-beta.0",
+    "@apollosproject/ui-notifications": "1.4.0-beta.0",
+    "@apollosproject/ui-onboarding": "1.4.0-beta.0",
+    "@apollosproject/ui-passes": "1.4.0-beta.0",
+    "@apollosproject/ui-scripture": "1.4.0-beta.0",
+    "@apollosproject/ui-storybook": "1.4.0-beta.0",
     "@react-native-community/async-storage": "^1.5.1",
     "@react-native-community/geolocation": "^2.0.2",
     "apollo-cache-inmemory": "1.5.1",
@@ -117,7 +117,7 @@
     "yup": "0.26.6"
   },
   "devDependencies": {
-    "@apollosproject/eslint-config": "^1.3.1",
+    "@apollosproject/eslint-config": "1.4.0-beta.0",
     "babel-eslint": "10.0.1",
     "codecov": "3.6.5",
     "eslint": "^4.19.1",
@@ -151,4 +151,4 @@
       "assets/fonts"
     ]
   }
-}
+}
\ No newline at end of file
diff --git a/apolloschurchapp/src/content-single/getContentItem.js b/apolloschurchapp/src/content-single/getContentItem.js
index 95695dd..40de3cb 100644
--- a/apolloschurchapp/src/content-single/getContentItem.js
+++ b/apolloschurchapp/src/content-single/getContentItem.js
@@ -6,9 +6,9 @@ export default gql`
     node(id: $itemId) {
       __typename
       ... on ContentItem {
-        ...contentItemFragment
+        ...contentCardFragment
       }
     }
   }
-  ${ApollosConfig.FRAGMENTS.CONTENT_ITEM_FRAGMENT}
+  ${ApollosConfig.FRAGMENTS.CONTENT_CARD_FRAGMENT}
 `;
diff --git a/apolloschurchapp/src/tabs/home/Features/ActionListFeature/ActionListFeature.stories.js b/apolloschurchapp/src/tabs/home/Features/ActionListFeature/ActionListFeature.stories.js
deleted file mode 100644
index e07dbea..0000000
--- a/apolloschurchapp/src/tabs/home/Features/ActionListFeature/ActionListFeature.stories.js
+++ /dev/null
@@ -1,172 +0,0 @@
-import React from 'react';
-import { storiesOf } from '@apollosproject/ui-storybook';
-
-import { BackgroundView, CenteredView } from '@apollosproject/ui-kit';
-import ActionListFeature from '.';
-
-const actions = [
-  {
-    id: 'fakeId1',
-    title: 'Boom',
-    subtitle: 'What',
-    parentChannel: {
-      id: '',
-      name: '',
-    },
-    image: {
-      sources: [
-        {
-          uri: 'https://picsum.photos/200/200',
-        },
-      ],
-    },
-  },
-  {
-    id: 'fakeId2',
-    title: 'Boom',
-    subtitle: 'What',
-    parentChannel: {
-      id: '',
-      name: '',
-    },
-    image: {
-      sources: [
-        {
-          uri: 'https://picsum.photos/200',
-        },
-      ],
-    },
-  },
-  {
-    id: 'fakeId3',
-    title: 'Boom',
-    subtitle: 'What',
-    parentChannel: {
-      id: '',
-      name: '',
-    },
-    image: {
-      sources: [
-        {
-          uri: 'https://picsum.photos/200',
-        },
-      ],
-    },
-  },
-  {
-    id: 'fakeId4',
-    title: 'Boom',
-    subtitle: 'What',
-    parentChannel: {
-      id: '',
-      name: '',
-    },
-    image: {
-      sources: [
-        {
-          uri: 'https://picsum.photos/200',
-        },
-      ],
-    },
-  },
-];
-
-storiesOf('ActionListFeature', module)
-  .addDecorator((story) => (
-    <BackgroundView>
-      {/* eslint-disable-next-line react-native/no-inline-styles */}
-      <CenteredView style={{ alignItems: 'stretch' }}>{story()}</CenteredView>
-    </BackgroundView>
-  ))
-  .add('example', () => (
-    <ActionListFeature
-      actions={actions}
-      title={'Title'}
-      subtitle={'Subtitle'}
-      onPressActionListButton={() => {}}
-    />
-  ))
-  .add('default', () => <ActionListFeature actions={actions} />)
-  .add('isLoading', () => {
-    const loadingStateData = [
-      {
-        id: 'fakeId1',
-        title: '',
-        subtitle: '',
-        parentChannel: {
-          id: '',
-          name: '',
-        },
-        image: {
-          sources: [
-            {
-              uri: '',
-            },
-          ],
-        },
-      },
-      {
-        id: 'fakeId2',
-        title: '',
-        subtitle: '',
-        parentChannel: {
-          id: '',
-          name: '',
-        },
-        image: {
-          sources: [
-            {
-              uri: '',
-            },
-          ],
-        },
-      },
-      {
-        id: 'fakeId3',
-        title: '',
-        subtitle: '',
-        parentChannel: {
-          id: '',
-          name: '',
-        },
-        image: {
-          sources: [
-            {
-              uri: '',
-            },
-          ],
-        },
-      },
-      {
-        id: 'fakeId4',
-        title: '',
-        subtitle: '',
-        parentChannel: {
-          id: '',
-          name: '',
-        },
-        image: {
-          sources: [
-            {
-              uri: '',
-            },
-          ],
-        },
-      },
-    ];
-
-    return (
-      <ActionListFeature
-        actions={loadingStateData}
-        onPressActionListButton={() => {}}
-        isLoading
-      />
-    );
-  })
-  .add('onPressActionListButton', () => (
-    <ActionListFeature actions={actions} onPressActionListButton={() => {}} />
-  ))
-  .add('subtitle', () => (
-    <ActionListFeature actions={actions} subtitle={'subtitle'} />
-  ))
-  .add('title', () => <ActionListFeature actions={actions} title={'Title'} />);
diff --git a/apolloschurchapp/src/tabs/home/Features/ActionListFeature/ActionListFeature.tests.js b/apolloschurchapp/src/tabs/home/Features/ActionListFeature/ActionListFeature.tests.js
deleted file mode 100644
index 37e554a..0000000
--- a/apolloschurchapp/src/tabs/home/Features/ActionListFeature/ActionListFeature.tests.js
+++ /dev/null
@@ -1,191 +0,0 @@
-import React from 'react';
-import renderer from 'react-test-renderer';
-
-import Providers from '../../../../Providers';
-
-import ActionListFeature from '.';
-
-const actions = [
-  {
-    id: 'fakeId1',
-    title: 'Boom',
-    subtitle: 'What',
-    parentChannel: {
-      id: '',
-      name: '',
-    },
-    image: {
-      sources: [
-        {
-          uri: 'https://picsum.photos/200/200',
-        },
-      ],
-    },
-  },
-  {
-    id: 'fakeId2',
-    title: 'Boom',
-    subtitle: 'What',
-    parentChannel: {
-      id: '',
-      name: '',
-    },
-    image: {
-      sources: [
-        {
-          uri: 'https://picsum.photos/200',
-        },
-      ],
-    },
-  },
-  {
-    id: 'fakeId3',
-    title: 'Boom',
-    subtitle: 'What',
-    parentChannel: {
-      id: '',
-      name: '',
-    },
-    image: {
-      sources: [
-        {
-          uri: 'https://picsum.photos/200',
-        },
-      ],
-    },
-  },
-  {
-    id: 'fakeId4',
-    title: 'Boom',
-    subtitle: 'What',
-    parentChannel: {
-      id: '',
-      name: '',
-    },
-    image: {
-      sources: [
-        {
-          uri: 'https://picsum.photos/200',
-        },
-      ],
-    },
-  },
-];
-
-describe('The ActionListFeatures component', () => {
-  it('should render', () => {
-    const tree = renderer.create(
-      <Providers>
-        <ActionListFeature actions={actions} />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render with a title', () => {
-    const tree = renderer.create(
-      <Providers>
-        <ActionListFeature
-          actions={actions}
-          title={'This renders smaller than its name would suggest'}
-        />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render with a subtitle', () => {
-    const tree = renderer.create(
-      <Providers>
-        <ActionListFeature
-          actions={actions}
-          subtitle={'This renders larger than you might expect'}
-        />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render a button for onPressActionListButton', () => {
-    const tree = renderer.create(
-      <Providers>
-        <ActionListFeature
-          actions={actions}
-          onPressActionListButton={jest.fn()}
-        />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render a loading state for isLoading', () => {
-    const loadingStateData = [
-      {
-        id: 'fakeId1',
-        title: '',
-        subtitle: '',
-        parentChannel: {
-          id: '',
-          name: '',
-        },
-        image: {
-          sources: [
-            {
-              uri: '',
-            },
-          ],
-        },
-      },
-      {
-        id: 'fakeId2',
-        title: '',
-        subtitle: '',
-        parentChannel: {
-          id: '',
-          name: '',
-        },
-        image: {
-          sources: [
-            {
-              uri: '',
-            },
-          ],
-        },
-      },
-      {
-        id: 'fakeId3',
-        title: '',
-        subtitle: '',
-        parentChannel: {
-          id: '',
-          name: '',
-        },
-        image: {
-          sources: [
-            {
-              uri: '',
-            },
-          ],
-        },
-      },
-      {
-        id: 'fakeId4',
-        title: '',
-        subtitle: '',
-        parentChannel: {
-          id: '',
-          name: '',
-        },
-        image: {
-          sources: [
-            {
-              uri: '',
-            },
-          ],
-        },
-      },
-    ];
-    const tree = renderer.create(
-      <Providers>
-        <ActionListFeature actions={loadingStateData} isLoading />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-});
diff --git a/apolloschurchapp/src/tabs/home/Features/ActionListFeature/index.js b/apolloschurchapp/src/tabs/home/Features/ActionListFeature/index.js
deleted file mode 100644
index 91a6eaa..0000000
--- a/apolloschurchapp/src/tabs/home/Features/ActionListFeature/index.js
+++ /dev/null
@@ -1,56 +0,0 @@
-import React, { memo } from 'react';
-import PropTypes from 'prop-types';
-
-import { styled, ActionListCard, H3, H6 } from '@apollosproject/ui-kit';
-
-const Title = styled(
-  ({ theme }) => ({
-    color: theme.colors.text.tertiary,
-  }),
-  'ActionListFeature.Title'
-)(H6);
-
-const Subtitle = styled({}, 'ActionListFeature.Subtitle')(H3);
-
-const ActionListFeature = memo(
-  ({
-    actions,
-    id,
-    isLoading,
-    onPressActionListButton,
-    onPressActionItem,
-    subtitle,
-    title,
-  }) => (
-    <ActionListCard
-      isLoading={isLoading}
-      key={id}
-      header={
-        <>
-          {isLoading || title ? ( // we check for isloading here so that they are included in the loading state
-            <Title numberOfLines={1}>{title}</Title>
-          ) : null}
-          {isLoading || subtitle ? <Subtitle>{subtitle}</Subtitle> : null}
-        </>
-      }
-      actions={actions}
-      onPressActionItem={onPressActionItem}
-      onPressActionListButton={onPressActionListButton}
-    />
-  )
-);
-
-ActionListFeature.displayName = 'Features';
-
-ActionListFeature.propTypes = {
-  // TODO: refactor ActionListCard to safely render without an actions array.
-  actions: PropTypes.arrayOf(PropTypes.shape({})).isRequired, // at least for the time being this is required
-  id: PropTypes.number,
-  isLoading: PropTypes.bool,
-  onPressActionListButton: PropTypes.func,
-  onPressActionItem: PropTypes.func,
-  subtitle: PropTypes.string,
-  title: PropTypes.string,
-};
-
-export default ActionListFeature;
diff --git a/apolloschurchapp/src/tabs/home/Features/CampaignItemListFeature/CampaignItemListFeature.stories.js b/apolloschurchapp/src/tabs/home/Features/CampaignItemListFeature/CampaignItemListFeature.stories.js
deleted file mode 100644
index e8595c8..0000000
--- a/apolloschurchapp/src/tabs/home/Features/CampaignItemListFeature/CampaignItemListFeature.stories.js
+++ /dev/null
@@ -1,50 +0,0 @@
-import React from 'react';
-import { storiesOf } from '@apollosproject/ui-storybook';
-
-import { BackgroundView, CenteredView } from '@apollosproject/ui-kit';
-import CampaignItemListFeature from '.';
-
-const cards = [
-  {
-    action: 'READ_CONTENT',
-    title: 'Live for Freedom Toolkit',
-    hasAction: null,
-    actionIcon: null,
-    labelText: null,
-    summary:
-      "Celebrate your freedom in Christ — in the car, shower, cubicle or wherever you go — with this playlist of NewSpring Worship's favorite songs of the summer.",
-    coverImage: [
-      {
-        uri: 'https://picsum.photos/800',
-      },
-    ],
-    relatedNode: {
-      id: 'UniversalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
-      __typename: 'UniversalContentItem',
-    },
-    __typename: 'CardListItem',
-  },
-];
-
-storiesOf('CampaignItemListFeature', module)
-  .addDecorator((story) => (
-    <BackgroundView>
-      {/* eslint-disable-next-line react-native/no-inline-styles */}
-      <CenteredView style={{ alignItems: 'stretch' }}>{story()}</CenteredView>
-    </BackgroundView>
-  ))
-  .add('example', () => (
-    <CampaignItemListFeature
-      cards={cards}
-      title={'Title'}
-      subtitle={'Subtitle'}
-    />
-  ))
-  .add('default', () => <CampaignItemListFeature cards={cards} />)
-  .add('isLoading', () => <CampaignItemListFeature cards={[]} isLoading />)
-  .add('subtitle', () => (
-    <CampaignItemListFeature cards={cards} subtitle={'subtitle'} />
-  ))
-  .add('title', () => (
-    <CampaignItemListFeature cards={cards} title={'Title'} />
-  ));
diff --git a/apolloschurchapp/src/tabs/home/Features/CampaignItemListFeature/CampaignItemListFeature.tests.js b/apolloschurchapp/src/tabs/home/Features/CampaignItemListFeature/CampaignItemListFeature.tests.js
deleted file mode 100644
index b332765..0000000
--- a/apolloschurchapp/src/tabs/home/Features/CampaignItemListFeature/CampaignItemListFeature.tests.js
+++ /dev/null
@@ -1,68 +0,0 @@
-import React from 'react';
-import renderer from 'react-test-renderer';
-
-import Providers from '../../../../Providers';
-
-import CampaignItemListFeature from '.';
-
-const cards = [
-  {
-    action: 'READ_CONTENT',
-    title: 'Card Title',
-    hasAction: null,
-    actionIcon: null,
-    labelText: null,
-    summary: 'Card summary',
-    coverImage: [
-      {
-        uri: 'https://picsum.photos/800',
-      },
-    ],
-    relatedNode: {
-      id: 'UniversalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
-      __typename: 'UniversalContentItem',
-    },
-    __typename: 'CardListItem',
-  },
-];
-
-describe('The CampaignItemListFeature component', () => {
-  it('should render', () => {
-    const tree = renderer.create(
-      <Providers>
-        <CampaignItemListFeature cards={cards} />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render a section title', () => {
-    const tree = renderer.create(
-      <Providers>
-        <CampaignItemListFeature
-          cards={cards}
-          title={'This renders smaller than its name would suggest'}
-        />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render with a subtitle', () => {
-    const tree = renderer.create(
-      <Providers>
-        <CampaignItemListFeature
-          cards={cards}
-          subtitle={'This renders larger than you might expect'}
-        />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render a loading state for isLoading', () => {
-    const tree = renderer.create(
-      <Providers>
-        <CampaignItemListFeature cards={[]} isLoading />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-});
diff --git a/apolloschurchapp/src/tabs/home/Features/CampaignItemListFeature/index.js b/apolloschurchapp/src/tabs/home/Features/CampaignItemListFeature/index.js
deleted file mode 100644
index bba130d..0000000
--- a/apolloschurchapp/src/tabs/home/Features/CampaignItemListFeature/index.js
+++ /dev/null
@@ -1,90 +0,0 @@
-import React, { memo } from 'react';
-import { View } from 'react-native';
-import PropTypes from 'prop-types';
-
-import {
-  FeaturedCard,
-  FeedView,
-  H2,
-  H5,
-  PaddedView,
-  styled,
-  withIsLoading,
-} from '@apollosproject/ui-kit';
-
-const Title = styled(
-  ({ theme }) => ({
-    color: theme.colors.text.tertiary,
-  }),
-  'CampaignItemListFeature.Title'
-)(H5);
-
-const Subtitle = styled({}, 'CampaignItemListFeature.Subtitle')(H2);
-
-const Header = styled(({ theme }) => ({
-  paddingTop: theme.sizing.baseUnit * 3,
-  paddingBottom: theme.sizing.baseUnit * 0.5,
-}))(PaddedView);
-
-// const getContent = ({ cards, isLoading }) => {
-//   let content = [];
-//   if (isLoading && !cards.length) {
-//     content = [
-//       {
-//         id: 'fakeId0',
-//         isLoading: true,
-//         title: 'Test',
-//         summary: 'Boom',
-//         channelType: '',
-//         coverImage: [],
-//         parentChannel: {
-//           id: '',
-//           name: '',
-//         },
-//       },
-//     ];
-//   } else {
-//     content = cards.map((card) => ({
-//       ...card,
-//       coverImage: get(card, 'coverImage.sources', undefined),
-//       __typename: card.relatedNode.__typename,
-//     }));
-//   }
-//
-//   return content;
-// };
-
-const CampaignItemListFeature = memo(
-  ({ cards, isLoading, listKey, onPressItem, subtitle, title }) => (
-    <View>
-      {isLoading || title || subtitle ? ( // only display the Header if we are loading or have a title/subtitle
-        <Header vertical={false}>
-          {isLoading || title ? ( // we check for isloading here so that they are included in the loading state
-            <Title numberOfLines={1}>{title}</Title>
-          ) : null}
-          {isLoading || subtitle ? <Subtitle>{subtitle}</Subtitle> : null}
-        </Header>
-      ) : null}
-      <FeedView
-        onPressItem={onPressItem}
-        ListItemComponent={FeaturedCard}
-        content={cards} // {getContent({ cards, isLoading })}
-        isLoading={isLoading}
-        listKey={listKey}
-      />
-    </View>
-  )
-);
-
-CampaignItemListFeature.displayName = 'Features';
-
-CampaignItemListFeature.propTypes = {
-  cards: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
-  isLoading: PropTypes.bool,
-  listKey: PropTypes.string,
-  onPressItem: PropTypes.func,
-  subtitle: PropTypes.string,
-  title: PropTypes.string,
-};
-
-export default withIsLoading(CampaignItemListFeature);
diff --git a/apolloschurchapp/src/tabs/home/Features/HorizontalCardListFeature/HorizontalCardListFeature.stories.js b/apolloschurchapp/src/tabs/home/Features/HorizontalCardListFeature/HorizontalCardListFeature.stories.js
deleted file mode 100644
index ef1f80e..0000000
--- a/apolloschurchapp/src/tabs/home/Features/HorizontalCardListFeature/HorizontalCardListFeature.stories.js
+++ /dev/null
@@ -1,82 +0,0 @@
-import React from 'react';
-import { storiesOf } from '@apollosproject/ui-storybook';
-
-import { BackgroundView, CenteredView } from '@apollosproject/ui-kit';
-import HorizontalCardListFeature from '.';
-
-const cards = [
-  {
-    action: 'READ_CONTENT',
-    title: 'Live for Freedom Toolkit',
-    hyphenatedTitle: 'Live for Freedom Toolkit',
-    hasAction: true,
-    actionIcon: 'umbrella',
-    labelText: 'Stormy',
-    summary:
-      "Celebrate your freedom in Christ — in the car, shower, cubicle or wherever you go — with this playlist of NewSpring Worship's favorite songs of the summer.",
-    coverImage: [
-      {
-        uri: 'https://picsum.photos/800',
-      },
-    ],
-    __typename: 'WeekendContentItem',
-    id: 'WeekendContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
-  },
-  {
-    action: 'READ_CONTENT',
-    title: 'Live for Freedom Toolkit',
-    hyphenatedTitle: 'Live for Freedom Toolkit',
-    hasAction: null,
-    actionIcon: null,
-    labelText: null,
-    summary:
-      "Celebrate your freedom in Christ — in the car, shower, cubicle or wherever you go — with this playlist of NewSpring Worship's favorite songs of the summer.",
-    coverImage: [
-      {
-        uri: 'https://picsum.photos/800',
-      },
-    ],
-    __typename: 'UniversalContentItem',
-    id: 'UniversalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
-  },
-  {
-    action: 'READ_CONTENT',
-    title: 'Live for Freedom Toolkit',
-    hyphenatedTitle: 'Live for Freedom Toolkit',
-    hasAction: null,
-    actionIcon: null,
-    labelText: null,
-    summary:
-      "Celebrate your freedom in Christ — in the car, shower, cubicle or wherever you go — with this playlist of NewSpring Worship's favorite songs of the summer.",
-    coverImage: [
-      {
-        uri: 'https://picsum.photos/800',
-      },
-    ],
-    __typename: 'DevotionalContentItem',
-    id: 'DevotionalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
-  },
-];
-
-storiesOf('HorizontalCardListFeature', module)
-  .addDecorator((story) => (
-    <BackgroundView>
-      {/* eslint-disable-next-line react-native/no-inline-styles */}
-      <CenteredView style={{ alignItems: 'stretch' }}>{story()}</CenteredView>
-    </BackgroundView>
-  ))
-  .add('example', () => (
-    <HorizontalCardListFeature
-      cards={cards}
-      title={'Title'}
-      subtitle={'Subtitle'}
-    />
-  ))
-  .add('default', () => <HorizontalCardListFeature cards={cards} />)
-  .add('isLoading', () => <HorizontalCardListFeature cards={[]} isLoading />)
-  .add('subtitle', () => (
-    <HorizontalCardListFeature cards={cards} subtitle={'subtitle'} />
-  ))
-  .add('title', () => (
-    <HorizontalCardListFeature cards={cards} title={'Title'} />
-  ));
diff --git a/apolloschurchapp/src/tabs/home/Features/HorizontalCardListFeature/HorizontalCardListFeature.tests.js b/apolloschurchapp/src/tabs/home/Features/HorizontalCardListFeature/HorizontalCardListFeature.tests.js
deleted file mode 100644
index 94e86d2..0000000
--- a/apolloschurchapp/src/tabs/home/Features/HorizontalCardListFeature/HorizontalCardListFeature.tests.js
+++ /dev/null
@@ -1,101 +0,0 @@
-import React from 'react';
-import renderer from 'react-test-renderer';
-
-import Providers from '../../../../Providers';
-
-import HorizontalCardListFeature from '.';
-
-const cards = [
-  {
-    action: 'READ_CONTENT',
-    title: 'Live for Freedom Toolkit',
-    hyphenatedTitle: 'Live for Freedom Toolkit',
-    hasAction: true,
-    actionIcon: 'umbrella',
-    labelText: 'Stormy',
-    summary:
-      "Celebrate your freedom in Christ — in the car, shower, cubicle or wherever you go — with this playlist of NewSpring Worship's favorite songs of the summer.",
-    coverImage: [
-      {
-        uri: 'https://picsum.photos/800',
-      },
-    ],
-    __typename: 'WeekendContentItem',
-    id: 'WeekendContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
-  },
-  {
-    action: 'READ_CONTENT',
-    title: 'Live for Freedom Toolkit',
-    hyphenatedTitle: 'Live for Freedom Toolkit',
-    hasAction: null,
-    actionIcon: null,
-    labelText: null,
-    summary:
-      "Celebrate your freedom in Christ — in the car, shower, cubicle or wherever you go — with this playlist of NewSpring Worship's favorite songs of the summer.",
-    coverImage: [
-      {
-        uri: 'https://picsum.photos/800',
-      },
-    ],
-    __typename: 'UniversalContentItem',
-    id: 'UniversalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
-  },
-  {
-    action: 'READ_CONTENT',
-    title: 'Live for Freedom Toolkit',
-    hyphenatedTitle: 'Live for Freedom Toolkit',
-    hasAction: null,
-    actionIcon: null,
-    labelText: null,
-    summary:
-      "Celebrate your freedom in Christ — in the car, shower, cubicle or wherever you go — with this playlist of NewSpring Worship's favorite songs of the summer.",
-    coverImage: [
-      {
-        uri: 'https://picsum.photos/800',
-      },
-    ],
-    __typename: 'DevotionalContentItem',
-    id: 'DevotionalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
-  },
-];
-
-describe('The HorizontalCardListFeature component', () => {
-  it('should render', () => {
-    const tree = renderer.create(
-      <Providers>
-        <HorizontalCardListFeature cards={cards} />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render a section title', () => {
-    const tree = renderer.create(
-      <Providers>
-        <HorizontalCardListFeature
-          cards={cards}
-          title={'This renders smaller than its name would suggest'}
-        />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render with a section subtitle', () => {
-    const tree = renderer.create(
-      <Providers>
-        <HorizontalCardListFeature
-          cards={cards}
-          subtitle={'This renders larger than you might expect'}
-        />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render a loading state for isLoading', () => {
-    const tree = renderer.create(
-      <Providers>
-        <HorizontalCardListFeature cards={[]} isLoading />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-});
diff --git a/apolloschurchapp/src/tabs/home/Features/HorizontalCardListFeature/index.js b/apolloschurchapp/src/tabs/home/Features/HorizontalCardListFeature/index.js
deleted file mode 100644
index d628382..0000000
--- a/apolloschurchapp/src/tabs/home/Features/HorizontalCardListFeature/index.js
+++ /dev/null
@@ -1,84 +0,0 @@
-import React, { PureComponent } from 'react';
-import { View } from 'react-native';
-import PropTypes from 'prop-types';
-
-import {
-  H2,
-  H5,
-  HorizontalTileFeed,
-  PaddedView,
-  styled,
-  TouchableScale,
-  withIsLoading,
-} from '@apollosproject/ui-kit';
-import { horizontalContentCardComponentMapper } from '@apollosproject/ui-connected';
-
-const Title = styled(
-  ({ theme }) => ({
-    color: theme.colors.text.tertiary,
-  }),
-  'HorizontalCardListFeature.Title'
-)(H5);
-
-const Subtitle = styled({}, 'HorizontalCardListFeature.Subtitle')(H2);
-
-const Header = styled(({ theme }) => ({
-  paddingTop: theme.sizing.baseUnit * 3,
-  paddingBottom: theme.sizing.baseUnit * 0.5,
-}))(PaddedView);
-
-class HorizontalCardListFeature extends PureComponent {
-  loadingStateObject = {
-    id: 'fakeId0',
-    isLoading: true,
-    title: 'Boom',
-    hasAction: true,
-    actionIcon: 'umbrella',
-    channelType: '',
-    coverImage: [],
-    // We need to assume a typename so horizontalContentCardComponentMapper knows what "type" to render
-    __typename: 'MediaContentItem',
-  };
-
-  static propTypes = {
-    cards: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
-    isLoading: PropTypes.bool,
-    listKey: PropTypes.string, // needed if multiple lists/feeds are displayed as siblings
-    onPressItem: PropTypes.func,
-    subtitle: PropTypes.string,
-    title: PropTypes.string,
-  };
-
-  keyExtractor = (item) => item && item.id;
-
-  renderItem = ({ item }) => (
-    <TouchableScale onPress={() => this.props.onPressItem(item)}>
-      {horizontalContentCardComponentMapper({ ...item })}
-    </TouchableScale>
-  );
-
-  render() {
-    return (
-      <View>
-        <Header vertical={false}>
-          {this.props.isLoading || this.props.title ? ( // we check for isloading here so that they are included in the loading state
-            <Title numberOfLines={1}>{this.props.title}</Title>
-          ) : null}
-          {this.props.isLoading || this.props.subtitle ? (
-            <Subtitle>{this.props.subtitle}</Subtitle>
-          ) : null}
-        </Header>
-        <HorizontalTileFeed
-          content={this.props.cards}
-          isLoading={this.props.isLoading}
-          listKey={this.props.listKey}
-          keyExtractor={this.keyExtractor}
-          loadingStateObject={this.loadingStateObject}
-          renderItem={this.renderItem}
-        />
-      </View>
-    );
-  }
-}
-
-export default withIsLoading(HorizontalCardListFeature);
diff --git a/apolloschurchapp/src/tabs/home/Features/VerticalCardListFeature/VerticalCardListFeature.stories.js b/apolloschurchapp/src/tabs/home/Features/VerticalCardListFeature/VerticalCardListFeature.stories.js
deleted file mode 100644
index eeeb895..0000000
--- a/apolloschurchapp/src/tabs/home/Features/VerticalCardListFeature/VerticalCardListFeature.stories.js
+++ /dev/null
@@ -1,50 +0,0 @@
-import React from 'react';
-import { storiesOf } from '@apollosproject/ui-storybook';
-
-import { BackgroundView, CenteredView } from '@apollosproject/ui-kit';
-import VerticalCardListFeature from '.';
-
-const cards = [
-  {
-    action: 'READ_CONTENT',
-    title: 'Live for Freedom Toolkit',
-    hasAction: null,
-    actionIcon: null,
-    labelText: null,
-    summary:
-      "Celebrate your freedom in Christ — in the car, shower, cubicle or wherever you go — with this playlist of NewSpring Worship's favorite songs of the summer.",
-    coverImage: [
-      {
-        uri: 'https://picsum.photos/800',
-      },
-    ],
-    relatedNode: {
-      id: 'UniversalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
-      __typename: 'UniversalContentItem',
-    },
-    __typename: 'CardListItem',
-  },
-];
-
-storiesOf('VerticalCardListFeature', module)
-  .addDecorator((story) => (
-    <BackgroundView>
-      {/* eslint-disable-next-line react-native/no-inline-styles */}
-      <CenteredView style={{ alignItems: 'stretch' }}>{story()}</CenteredView>
-    </BackgroundView>
-  ))
-  .add('example', () => (
-    <VerticalCardListFeature
-      cards={cards}
-      title={'Title'}
-      subtitle={'Subtitle'}
-    />
-  ))
-  .add('default', () => <VerticalCardListFeature cards={cards} />)
-  .add('isLoading', () => <VerticalCardListFeature cards={[]} isLoading />)
-  .add('subtitle', () => (
-    <VerticalCardListFeature cards={cards} subtitle={'subtitle'} />
-  ))
-  .add('title', () => (
-    <VerticalCardListFeature cards={cards} title={'Title'} />
-  ));
diff --git a/apolloschurchapp/src/tabs/home/Features/VerticalCardListFeature/VerticalCardListFeature.tests.js b/apolloschurchapp/src/tabs/home/Features/VerticalCardListFeature/VerticalCardListFeature.tests.js
deleted file mode 100644
index fa7a89f..0000000
--- a/apolloschurchapp/src/tabs/home/Features/VerticalCardListFeature/VerticalCardListFeature.tests.js
+++ /dev/null
@@ -1,68 +0,0 @@
-import React from 'react';
-import renderer from 'react-test-renderer';
-
-import Providers from '../../../../Providers';
-
-import VerticalCardListFeature from '.';
-
-const cards = [
-  {
-    action: 'READ_CONTENT',
-    title: 'Card Title',
-    hasAction: null,
-    actionIcon: null,
-    labelText: null,
-    summary: 'Card summary',
-    coverImage: [
-      {
-        uri: 'https://picsum.photos/800',
-      },
-    ],
-    relatedNode: {
-      id: 'UniversalContentItem:95ff79f60a028b1b506aaeedf8b4c6ae',
-      __typename: 'UniversalContentItem',
-    },
-    __typename: 'CardListItem',
-  },
-];
-
-describe('The VerticalCardListFeature component', () => {
-  it('should render', () => {
-    const tree = renderer.create(
-      <Providers>
-        <VerticalCardListFeature cards={cards} />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render a section title', () => {
-    const tree = renderer.create(
-      <Providers>
-        <VerticalCardListFeature
-          cards={cards}
-          title={'This renders smaller than its name would suggest'}
-        />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render with a subtitle', () => {
-    const tree = renderer.create(
-      <Providers>
-        <VerticalCardListFeature
-          cards={cards}
-          subtitle={'This renders larger than you might expect'}
-        />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render a loading state for isLoading', () => {
-    const tree = renderer.create(
-      <Providers>
-        <VerticalCardListFeature cards={[]} isLoading />
-      </Providers>
-    );
-    expect(tree).toMatchSnapshot();
-  });
-});
diff --git a/apolloschurchapp/src/tabs/home/Features/VerticalCardListFeature/index.js b/apolloschurchapp/src/tabs/home/Features/VerticalCardListFeature/index.js
deleted file mode 100644
index bfa25ed..0000000
--- a/apolloschurchapp/src/tabs/home/Features/VerticalCardListFeature/index.js
+++ /dev/null
@@ -1,88 +0,0 @@
-import React, { memo } from 'react';
-import { View } from 'react-native';
-import PropTypes from 'prop-types';
-
-import {
-  FeedView,
-  H2,
-  H5,
-  PaddedView,
-  styled,
-  withIsLoading,
-} from '@apollosproject/ui-kit';
-import { contentCardComponentMapper } from '@apollosproject/ui-connected';
-
-const Title = styled(
-  ({ theme }) => ({
-    color: theme.colors.text.tertiary,
-  }),
-  'VerticalCardListFeature.Title'
-)(H5);
-
-const Subtitle = styled({}, 'VerticalCardListFeature.Subtitle')(H2);
-
-const Header = styled(({ theme }) => ({
-  paddingTop: theme.sizing.baseUnit * 3,
-  paddingBottom: theme.sizing.baseUnit * 0.5,
-}))(PaddedView);
-
-// const getContent = ({ cards, isLoading }) => {
-//   let content = [];
-//   if (isLoading && !cards.length) {
-//     content = [
-//       {
-//         id: 'fakeId0',
-//         isLoading: true,
-//         title: 'Test',
-//         summary: 'Boom',
-//         channelType: '',
-//         coverImage: [],
-//         parentChannel: {
-//           id: '',
-//           name: '',
-//         },
-//       },
-//     ];
-//   } else {
-//     content = cards.map((card) => ({
-//       ...card,
-//       coverImage: get(card, 'coverImage.sources', undefined),
-//       __typename: card.relatedNode.__typename,
-//     }));
-//   }
-//
-//   return content;
-// };
-
-const VerticalCardListFeature = memo(
-  ({ cards, isLoading, listKey, onPressItem, subtitle, title }) => (
-    <View>
-      <Header vertical={false}>
-        {isLoading || title ? ( // we check for isloading here so that they are included in the loading state
-          <Title numberOfLines={1}>{title}</Title>
-        ) : null}
-        {isLoading || subtitle ? <Subtitle>{subtitle}</Subtitle> : null}
-      </Header>
-      <FeedView
-        onPressItem={onPressItem}
-        ListItemComponent={contentCardComponentMapper}
-        content={cards} // {getContent({ cards, isLoading })}
-        isLoading={isLoading}
-        listKey={listKey}
-      />
-    </View>
-  )
-);
-
-VerticalCardListFeature.displayName = 'Features';
-
-VerticalCardListFeature.propTypes = {
-  cards: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
-  isLoading: PropTypes.bool,
-  listKey: PropTypes.string,
-  onPressItem: PropTypes.func,
-  subtitle: PropTypes.string,
-  title: PropTypes.string,
-};
-
-export default withIsLoading(VerticalCardListFeature);
diff --git a/apolloschurchapp/src/tabs/home/Features/getFeedFeatures.js b/apolloschurchapp/src/tabs/home/Features/getFeedFeatures.js
index bd25878..447dd68 100644
--- a/apolloschurchapp/src/tabs/home/Features/getFeedFeatures.js
+++ b/apolloschurchapp/src/tabs/home/Features/getFeedFeatures.js
@@ -1,71 +1,15 @@
 import gql from 'graphql-tag';
+import ApollosConfig from '@apollosproject/config';
 
 export default gql`
   query getFeedFeatures {
     userFeedFeatures {
-      ... on ActionListFeature {
-        id
-        title
-        subtitle
-        actions {
-          id
-          title
-          subtitle
-          action
-          image {
-            sources {
-              uri
-            }
-          }
-          relatedNode {
-            id
-          }
-        }
-      }
-      ... on VerticalCardListFeature {
-        id
-        isFeatured
-        title
-        subtitle
-        cards {
-          action
-          title
-          hasAction
-          actionIcon
-          labelText
-          summary
-          coverImage {
-            sources {
-              uri
-            }
-          }
-          relatedNode {
-            id
-          }
-        }
-      }
-      ... on HorizontalCardListFeature {
-        id
-        title
-        subtitle
-        cards {
-          action
-          title
-          hyphenatedTitle: title(hyphenated: true)
-          hasAction
-          actionIcon
-          labelText
-          summary
-          coverImage {
-            sources {
-              uri
-            }
-          }
-          relatedNode {
-            id
-          }
-        }
-      }
+      ...ActionListFeatureFragment
+      ...VerticalCardListFeatureFragment
+      ...HorizontalCardListFeatureFragment
     }
   }
+  ${ApollosConfig.FRAGMENTS.ACTION_LIST_FEATURE_FRAGMENT}
+  ${ApollosConfig.FRAGMENTS.VERTICAL_CARD_LIST_FEATURE_FRAGMENT}
+  ${ApollosConfig.FRAGMENTS.HORIZONTAL_CARD_LIST_FEATURE_FRAGMENT}
 `;
diff --git a/apolloschurchapp/src/tabs/home/Features/index.js b/apolloschurchapp/src/tabs/home/Features/index.js
index 7b072d3..099cd92 100644
--- a/apolloschurchapp/src/tabs/home/Features/index.js
+++ b/apolloschurchapp/src/tabs/home/Features/index.js
@@ -3,10 +3,12 @@ import { Query } from 'react-apollo';
 import { get } from 'lodash';
 // import PropTypes from 'prop-types';
 
-import ActionListFeature from './ActionListFeature';
-import CampaignItemListFeature from './CampaignItemListFeature';
-import HorizontalCardListFeature from './HorizontalCardListFeature';
-import VerticalCardListFeature from './VerticalCardListFeature';
+import {
+  ActionListFeature,
+  CampaignItemListFeature,
+  HorizontalCardListFeature,
+  VerticalCardListFeature,
+} from '@apollosproject/ui-connected';
 
 import GET_FEED_FEATURES from './getFeedFeatures';
 
